<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <title>
    
    Recology, R/etc.
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:300,400italic,400,600,700|Abril+Fatface">

  <link rel="stylesheet" href="/public/css/bootstrap/css/bootstrap.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/favicon.ico">
  <link rel="shortcut icon" href="/public/favicon.ico">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css" rel="stylesheet">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body class="theme-base-0f layout-reverse">

    <header class="masthead">
      <div class="masthead-inner">
        <h1>Recology</h1>
        <!-- <h1> <a href="http://recology.info/">Recology</a></h1> -->
        <p class="lead">R/etc.</p>

        <div class="colophon">
          <ul class="colophon-links">
            <li>
              <a href="/"><i class="fa fa-home fa-lg"></i></a>&nbsp;
              <a href="/about"><i class="fa fa-info-circle fa-lg"></i></a>&nbsp;
              <a href="/archives"><i class="fa fa-archive fa-lg"></i></a>&nbsp;
              <a href="/rresources"><i class="fa fa-book fa-lg"></i></a>&nbsp;
              <a href="http://rforcats.net/" rel><i class="fa fa-graduation-cap fa-lg"></i></a>&nbsp;
              <a href="/feed.xml"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
              <a href="https://twitter.com/sckottie"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
              <a href="/fork"><i class="fa fa-spinner fa-lg"></i></a>
            </li>
          </ul>
          <!-- <small><a href="https://github.com/mdo/hyde">Hyde</a> from <a href="https://twitter.com/mdo" target="_blank">@mdo</a>.</small> -->
        </div>
      </div>
    </header>

    <div class="content container">
      <div class="posts">
  <a style="float:right;" href="/archives" data-toggle="tooltip" data-placement="bottom" title="Archives"><i class="fa fa-archive fa-lg"></i></a>
  <a style="float:right;" href="/tags"><i class="fa fa-tags fa-lg"></i></a>&nbsp;
  
  <div class="post">
    <h1>
      <a href="/2012/12/taxize/">
        One R package for all your taxonomic needs
      </a>
    </h1>

    <span class="post-date">06 Dec 2012</span>

    UPDATE: there were some errors in the tests for `taxize`, so the binaries aren't avaiable yet. You can install from source though, see below. 


Getting taxonomic information for the set of species you are studying can be a pain in the ass. You have to manually type, or paste in, your species one-by-one. Or, if you are lucky, there is a web service in which you can upload a list of species. Encyclopedia of Life (EOL) has a service where you can do this [here](http://gni.globalnames.org/parsers/new). But is this reproducible? No. 

Getting your taxonomic information for your species can now be done programatically in R. Do you want to get taxonomic information from ITIS. We got that. Tropicos? We got that too. uBio? No worries, we got that. What about theplantlist.org? Yep, got that. Encyclopedia of Life? Indeed. What about getting sequence data for a taxon? Oh hell yeah, you can get sequences available for a taxon across all genes, or get all records for a taxon for a specific gene. 

Of course this is all possible because these data providers have open APIs so that we can facilitate your computer talking to their database. Fun! 

Why get your taxonomic data programatically? Because it's 1) faster than by hand in web sites/looking up in books, 2) reproducible, especially if you share your code (damnit!), and 3) you can easily mash up your new taxonomic data to get sequences to build a phylogeny, etc.

I'll give a few examples of using `taxize` based around use cases, that is, stuff someone might actually do instead of what particular functions do.

***************

### Install packages.  You can get from CRAN or GitHub. 

{% highlight r %}
# install.packages("ritis") # uncomment if not already installed
# install_github('taxize_', 'ropensci') # uncomment if not already installed
# install.packages("taxize", type="source") # uncomment if not already installed
library(ritis)
library(taxize)
{% endhighlight %}


***************

### Attach family names to a list of species. 

#### I often have a list of species that I studied and simply want to get their family names to, for example, make a table for the paper I'm writing.

{% highlight r %}
# For one species
itis_name(query = "Poa annua", get = "family")
{% endhighlight %}



{% highlight text %}

Retrieving data for species ' Poa annua '
{% endhighlight %}



{% highlight text %}
[1] "Poaceae"
{% endhighlight %}



{% highlight r %}

# For many species
species <- c("Poa annua", "Abies procera", "Helianthus annuus", "Coffea arabica")
famnames <- sapply(species, itis_name, get = "family", USE.NAMES = F)
{% endhighlight %}



{% highlight text %}

Retrieving data for species ' Poa annua '
{% endhighlight %}



{% highlight text %}

Retrieving data for species ' Abies procera '
{% endhighlight %}



{% highlight text %}

Retrieving data for species ' Helianthus annuus '
{% endhighlight %}



{% highlight text %}

Retrieving data for species ' Coffea arabica '
{% endhighlight %}



{% highlight r %}
data.frame(species = species, family = famnames)
{% endhighlight %}



{% highlight text %}
            species     family
1         Poa annua    Poaceae
2     Abies procera   Pinaceae
3 Helianthus annuus Asteraceae
4    Coffea arabica  Rubiaceae
{% endhighlight %}


***************

### Resolve taxonomic names. 

#### This is a common use case for ecologists/evolutionary biologists, or at least should be. That is, species names you have for your own data, or when using other's data, could be old names - and if you need the newest names for your species list, how can you make this as painless as possible? You can query taxonomic data from many different sources with `taxize`. 

{% highlight r %}
# The iPlantCollaborative provides access via API to their taxonomic name
# resolution service (TNRS)
mynames <- c("shorea robusta", "pandanus patina", "oryza sativa", "durio zibethinus", 
    "rubus ulmifolius", "asclepias curassavica", "pistacia lentiscus")
iplant_tnrsmatch(retrieve = "all", taxnames = c("helianthus annuus", "acacia", 
    "gossypium"), output = "names")
{% endhighlight %}



{% highlight text %}
       AcceptedName   MatchFam MatchGenus MatchScore    Accept?
1 Helianthus annuus Asteraceae Helianthus          1 No opinion
2            Acacia   Fabaceae     Acacia          1 No opinion
3                                  Acacia          1 No opinion
4         Gossypium  Malvaceae  Gossypium          1 No opinion
     SubmittedNames
1 helianthus annuus
2            acacia
3            acacia
4         gossypium
{% endhighlight %}



{% highlight r %}

# The global names resolver is another attempt at this, hitting many
# different data sources
gnr_resolve(names = c("Helianthus annuus", "Homo sapiens"), returndf = TRUE)
{% endhighlight %}



{% highlight text %}
   data_source_id    submitted_name       name_string score
1               4 Helianthus annuus Helianthus annuus 0.988
3              10 Helianthus annuus Helianthus annuus 0.988
5              12 Helianthus annuus Helianthus annuus 0.988
8             110 Helianthus annuus Helianthus annuus 0.988
11            159 Helianthus annuus Helianthus annuus 0.988
13            166 Helianthus annuus Helianthus annuus 0.988
15            169 Helianthus annuus Helianthus annuus 0.988
2               4      Homo sapiens      Homo sapiens 0.988
4              10      Homo sapiens      Homo sapiens 0.988
6              12      Homo sapiens      Homo sapiens 0.988
7             107      Homo sapiens      Homo sapiens 0.988
9             122      Homo sapiens      Homo sapiens 0.988
10            123      Homo sapiens      Homo sapiens 0.988
12            159      Homo sapiens      Homo sapiens 0.988
14            168      Homo sapiens      Homo sapiens 0.988
16            169      Homo sapiens      Homo sapiens 0.988
                     title
1                     NCBI
3                 Freebase
5                      EOL
8     Illinois Wildflowers
11                 CU*STAR
13                   nlbif
15           uBio NameBank
2                     NCBI
4                 Freebase
6                      EOL
7                AskNature
9                 BioPedia
10                   AnAge
12                 CU*STAR
14 Index to Organism Names
16           uBio NameBank
{% endhighlight %}



{% highlight r %}

# We can hit the Plantminer API too
plants <- c("Myrcia lingua", "Myrcia bella", "Ocotea pulchella", "Miconia", 
    "Coffea arabica var. amarella", "Bleh")
plantminer(plants)
{% endhighlight %}



{% highlight text %}
Myrcia lingua 
Myrcia bella 
Ocotea pulchella 
Miconia 
Coffea arabica var. amarella 
Bleh 
{% endhighlight %}



{% highlight text %}
              fam   genus                    sp             author
1       Myrtaceae  Myrcia                lingua   (O. Berg) Mattos
2       Myrtaceae  Myrcia                 bella           Cambess.
3       Lauraceae  Ocotea             pulchella (Nees & Mart.) Mez
4 Melastomataceae Miconia                    NA        Ruiz & Pav.
5       Rubiaceae  Coffea arabica var. amarella        A. Froehner
6              NA    Bleh                    NA                 NA
            source source.id   status confidence suggestion       database
1              TRO 100227036       NA         NA         NA       Tropicos
2             WCSP    131057 Accepted          H         NA The Plant List
3 WCSP (in review)    989758 Accepted          M         NA The Plant List
4              TRO  40018467       NA         NA         NA       Tropicos
5              TRO 100170231       NA         NA         NA       Tropicos
6               NA        NA       NA         NA       Baea             NA
{% endhighlight %}



{% highlight r %}

# We made a light wrapper around the Taxonstand package to search
# Theplantlist.org too
splist <- c("Heliathus annuus", "Abies procera", "Poa annua", "Platanus occidentalis", 
    "Carex abrupta", "Arctostaphylos canescens", "Ocimum basilicum", "Vicia faba", 
    "Quercus kelloggii", "Lactuca serriola")
tpl_search(taxon = splist)
{% endhighlight %}



{% highlight text %}
            Genus      Species Infraspecific Plant.Name.Index
1       Heliathus       annuus                          FALSE
2           Abies      procera                           TRUE
3             Poa        annua                           TRUE
4        Platanus occidentalis                           TRUE
5           Carex      abrupta                           TRUE
6  Arctostaphylos    canescens                           TRUE
7          Ocimum    basilicum                           TRUE
8           Vicia         faba                           TRUE
9         Quercus    kelloggii                           TRUE
10        Lactuca     serriola                           TRUE
   Taxonomic.status      Family      New.Genus  New.Species
1                                    Heliathus       annuus
2          Accepted    Pinaceae          Abies      procera
3          Accepted     Poaceae            Poa        annua
4          Accepted Platanaceae       Platanus occidentalis
5          Accepted  Cyperaceae          Carex      abrupta
6          Accepted   Ericaceae Arctostaphylos    canescens
7          Accepted   Lamiaceae         Ocimum    basilicum
8          Accepted Leguminosae          Vicia         faba
9          Accepted    Fagaceae        Quercus    kelloggii
10         Accepted  Compositae        Lactuca     serriola
   New.Infraspecific Authority  Typo WFormat
1                              FALSE   FALSE
2                       Rehder FALSE   FALSE
3                           L. FALSE   FALSE
4                           L. FALSE   FALSE
5               <NA>     Mack. FALSE   FALSE
6                       Eastw. FALSE   FALSE
7                           L. FALSE   FALSE
8                           L. FALSE   FALSE
9               <NA>     Newb. FALSE   FALSE
10                          L. FALSE   FALSE
{% endhighlight %}


***************

### Taxonomic hierarchies 

#### I often want the full taxonomic hierarchy for a set of species. That is, give me the family, order, class, etc. for my list of species. There are two different easy ways to do this with `taxize`. The first example uses EOL.

***************

#### Using EOL.

{% highlight r %}
pageid <- eol_search("Quercus douglasii")$id[1]  # first need to search for the taxon's page on EOL
out <- eol_pages(taxonconceptID = pageid)  # then we nee to get the taxon ID used by EOL

# Notice that there are multiple different sources you can pull the
# hierarchy from. Note even that you can get the hierarchy from the ITIS
# service via this EOL API.
out
{% endhighlight %}



{% highlight text %}
  identifier                 scientificName
1   46203061 Quercus douglasii Hook. & Arn.
2   48373995 Quercus douglasii Hook. & Arn.
                                  nameAccordingTo sourceIdentfier
1  Integrated Taxonomic Information System (ITIS)           19322
2 Species 2000 & ITIS Catalogue of Life: May 2012         9723391
  taxonRank
1   Species
2   Species
{% endhighlight %}



{% highlight r %}

# Then the hierarchy!
eol_hierarchy(out[out$nameAccordingTo == "Species 2000 & ITIS Catalogue of Life: May 2012", 
    "identifier"])
{% endhighlight %}



{% highlight text %}
  sourceIdentifier  taxonID parentNameUsageID taxonConceptID
1         11017504 48276627                 0            281
2         11017505 48276628          48276627            282
3         11017506 48276629          48276628            283
4         11022500 48373354          48276629           4184
5         11025284 48373677          48373354           4197
  scientificName taxonRank
1        Plantae   kingdom
2  Magnoliophyta    phylum
3  Magnoliopsida     class
4        Fagales     order
5       Fagaceae    family
{% endhighlight %}



{% highlight r %}
eol_hierarchy(out[out$nameAccordingTo == "Integrated Taxonomic Information System (ITIS)", 
    "identifier"])  # and from ITIS, slightly different than ITIS output below, which includes taxa all the way down.
{% endhighlight %}



{% highlight text %}
   sourceIdentifier  taxonID parentNameUsageID taxonConceptID
1            202422 46150613                 0            281
2            846492 46159776          46150613        8654492
3            846494 46161961          46159776       28818077
4            846496 46167532          46161961           4494
5            846504 46169010          46167532       28825126
6            846505 46169011          46169010            282
7             18063 46169012          46169011            283
8            846548 46202954          46169012       28859070
9             19273 46202955          46202954           4184
10            19275 46203022          46202955           4197
    scientificName     taxonRank
1          Plantae       kingdom
2   Viridaeplantae    subkingdom
3     Streptophyta  infrakingdom
4     Tracheophyta      division
5  Spermatophytina   subdivision
6     Angiospermae infradivision
7    Magnoliopsida         class
8          Rosanae    superorder
9          Fagales         order
10        Fagaceae        family
{% endhighlight %}


***************

#### And getting a taxonomic hierarchy using ITIS.

{% highlight r %}
# First, get the taxonomic serial number (TSN) that ITIS uses
mytsn <- get_tsn("Quercus douglasii", "sciname")
{% endhighlight %}



{% highlight text %}

Retrieving data for species ' Quercus douglasii '
{% endhighlight %}



{% highlight r %}

# Get the full taxonomic hierarchy for a taxon from the TSN
itis(mytsn, "getfullhierarchyfromtsn")
{% endhighlight %}



{% highlight text %}
$`1`
        parentName parentTsn      rankName         taxonName    tsn
1                                  Kingdom           Plantae 202422
2          Plantae    202422    Subkingdom    Viridaeplantae 846492
3   Viridaeplantae    846492  Infrakingdom      Streptophyta 846494
4     Streptophyta    846494      Division      Tracheophyta 846496
5     Tracheophyta    846496   Subdivision   Spermatophytina 846504
6  Spermatophytina    846504 Infradivision      Angiospermae 846505
7     Angiospermae    846505         Class     Magnoliopsida  18063
8    Magnoliopsida     18063    Superorder           Rosanae 846548
9          Rosanae    846548         Order           Fagales  19273
10         Fagales     19273        Family          Fagaceae  19275
11        Fagaceae     19275         Genus           Quercus  19276
12         Quercus     19276       Species Quercus douglasii  19322

{% endhighlight %}



{% highlight r %}

# But this can be even easier!
classification(get_tsn("Quercus douglasii"))  # Boom!
{% endhighlight %}



{% highlight text %}

Retrieving data for species ' Quercus douglasii '
{% endhighlight %}



{% highlight text %}
$`1`
        parentName parentTsn      rankName         taxonName    tsn
1                                  Kingdom           Plantae 202422
2          Plantae    202422    Subkingdom    Viridaeplantae 846492
3   Viridaeplantae    846492  Infrakingdom      Streptophyta 846494
4     Streptophyta    846494      Division      Tracheophyta 846496
5     Tracheophyta    846496   Subdivision   Spermatophytina 846504
6  Spermatophytina    846504 Infradivision      Angiospermae 846505
7     Angiospermae    846505         Class     Magnoliopsida  18063
8    Magnoliopsida     18063    Superorder           Rosanae 846548
9          Rosanae    846548         Order           Fagales  19273
10         Fagales     19273        Family          Fagaceae  19275
11        Fagaceae     19275         Genus           Quercus  19276
12         Quercus     19276       Species Quercus douglasii  19322

{% endhighlight %}



{% highlight r %}

# You can also do this easy-peasy route to a taxonomic hierarchy using
# uBio
classification(get_uid("Ornithorhynchus anatinus"))
{% endhighlight %}



{% highlight text %}
$`1`
       ScientificName         Rank    UID
1  cellular organisms      no rank 131567
2           Eukaryota superkingdom   2759
3        Opisthokonta      no rank  33154
4             Metazoa      kingdom  33208
5           Eumetazoa      no rank   6072
6           Bilateria      no rank  33213
7           Coelomata      no rank  33316
8       Deuterostomia      no rank  33511
9            Chordata       phylum   7711
10           Craniata    subphylum  89593
11         Vertebrata      no rank   7742
12      Gnathostomata   superclass   7776
13         Teleostomi      no rank 117570
14       Euteleostomi      no rank 117571
15      Sarcopterygii      no rank   8287
16          Tetrapoda      no rank  32523
17            Amniota      no rank  32524
18           Mammalia        class  40674
19        Prototheria      no rank   9254
20        Monotremata        order   9255
21  Ornithorhynchidae       family   9256
22    Ornithorhynchus        genus   9257

{% endhighlight %}


***************

### Sequences?

#### While you are at doing taxonomic stuff, you often wonder "hmmm, I wonder if there are any sequence data available for my species?" So, you can use `get_seqs` to search for specific genes for a species, and `get_genes_avail` to find out what genes are available for a certain species. These functions search for data on [NCBI](http://www.ncbi.nlm.nih.gov/).

{% highlight r %}
# Get sequences (sequence is provied in output, but hiding here for
# brevity). What's nice about this is that it gets the longest sequence
# avaialable for the gene you searched for, and if there isn't anything
# available, it lets you get a sequence from a congener if you set
# getrelated=TRUE. The last column in the output data.frame also tells you
# what species the sequence is from.
out <- get_seqs(taxon_name = "Acipenser brevirostrum", gene = c("5S rRNA"), 
    seqrange = "1:3000", getrelated = T, writetodf = F)
out[, !names(out) %in% "sequence"]
{% endhighlight %}



{% highlight text %}
                   taxon                                         gene_desc
1 Acipenser brevirostrum Acipenser brevirostrum 5S rRNA gene, clone BRE92A
     gi_no     acc_no length                 spused
1 60417159 AJ745069.1    121 Acipenser brevirostrum
{% endhighlight %}



{% highlight r %}

# Search for available sequences
out <- get_genes_avail(taxon_name = "Umbra limi", seqrange = "1:2000", getrelated = F)
out[grep("RAG1", out$genesavail, ignore.case = T), ]  # does the string 'RAG1' exist in any of the gene names
{% endhighlight %}



{% highlight text %}
        spused length
414 Umbra limi    732
427 Umbra limi    959
434 Umbra limi   1631
                                                                            genesavail
414 isolate UlimA recombinase activating protein 1 (rag1) gene, exon 3 and partial cds
427           recombination-activating protein 1 (RAG1) gene, intron 2 and partial cds
434                        recombination-activating protein 1 (RAG1) gene, partial cds
    predicted
414  JX190826
427  AY459526
434  AY380548
{% endhighlight %}



*********
#### Get the .Rmd file used to create this post [at my github account](https://github.com/sckott/sckott.github.com/tree/master/_drafts/2012-12-06-taxize.Rmd) - or [.md file](https://github.com/sckott/sckott.github.com/tree/master/_posts/2012-12-06-taxize.md).

#### Written in [Markdown](http://daringfireball.net/projects/markdown/), with help from [knitr](http://yihui.name/knitr/).

  </div>
  
  <div class="post">
    <h1>
      <a href="/2012/11/altecology/">
        Altecology, a call to unconference action
      </a>
    </h1>

    <span class="post-date">15 Nov 2012</span>

    Note: This post is cross-posted on Sandra Chung's blog [here](http://sandrachung.com/).

*********

### The rise of the unconference
The Ecological Society of America meeting is holding its [98th annual meeting](http://www.esa.org/minneapolis/) next year in Minneapolis, MN. Several thousand students and professionals in ecological science and education will gather to hear and read the latest work and ideas in ecology in the familiar poster and lecture formats that are the core of every major scientific conference. But a subset of these people will get a taste of something a little bit different: an unconference within the conference.

The most important difference between traditional science conferences and the unconference format is that it prioritizes human interaction. Often the best and most important parts of science meetings are the interactions between talks, next to posters, and at the end of the day over drinks. These connections pave the way for collaborations and friendships that nourish our professional and personal lives with shared opportunities, camaraderie and support. 

In recognition of the increasing relative importance of the “meeting” part of a science meeting, the unconference format emphasizes interaction over presentation. It attempts to engage participants to the maximum extent reasonable in discussion and doing. [Science Online](http://scienceonline.com/) is a good example of this unconference format, in which the session topics are typically decided on democratically by the conference attendees (partly before arrival, partly on arrival), and you vote with your feet by going to and leaving sessions as you desire. 

*********

### Ecology is changing
Ecology is now adopting some of the same online and social tools that are already accelerating innovation in computing and other science disciplines. Ecologists, ecology students and educators are asking many of the same basic questions they have always asked: What should we be doing? How do we do it better and faster? Social media, open source software, open science, altmetrics, crowdsourcing, crowdfunding, data visualization, data sharing, alternative peer review, and an increasing emphasis on more and better communication and collaboration are just some of the newer tools being put forth to help address those questions in the 21st century.

Social media is rapidly becoming more common in ecologists’ toolkits to disseminate news of their new papers, communicate about research and research tools, and even filter the deluge of publications. Tools like blogs, Twitter and Facebook are filling the communication gaps between annual meetings and adding a new layer of conversation and connection to conferences and classrooms.

Social media, in turn, is connecting scientists directly to people besides their immediate colleagues who appreciate the impact of their work and want it to continue. The crowdfunding movement - exemplified by Kickstarter - has spurred similar alternative science funding projects such as SciFund. #SciFund project has shown that social media engagement increases donations to crowdfunded research ([interview with Jarrett Byrnes](http://jecology.libsyn.com/interview-with-jarrett-byrnes-on-science-crowdfunding)). 

In addition, we are in the era of big data, and [this includes ecology](http://www.neoninc.org/news/big-data-part-i). To deal with this “data deluge”, ecologists increasingly want to learn how to manage, share, retrieve, use, and visualize data. There are new tools for all of these tasks - we need to aid each other in learning them. Online and offline communities are coalescing around the development and dissemination of these tools for the benefit of ecological science, and they are meeting face-to-face at our ecological unconference.

Science in general is becoming increasingly complex and calling for larger and larger collaborations. This growth in turn is spurring a drive toward more openness and transparency within the culture of science. The more collaborative and complex scientific study becomes, the more scientists depend upon each other to do good work that we can all build upon with confidence. The often unstated assumption about ecology, and all of science, is that research findings are reproducible; but that assumption is quite shaky given the increasing number of retractions (see the [Retraction Watch blog](http://retractionwatch.wordpress.com/)) and findings that much research is not reproducible (see media coverage [here](http://www.nature.com/nature/journal/v483/n7391/full/483531a.html) and [here](http://www.reuters.com/article/2012/03/28/us-science-cancer-idUSBRE82R12P20120328)). 

A recent initiative seeks to facilitate attempts to reproduce research: [The Reproducibility Initiative](https://www.scienceexchange.com/reproducibility). Jarrett Byrnes [spoke at #ESA2012](http://www.slideshare.net/JarrettByrnes/taking-the-ecological-conversation-online) of how transparent, online discourse enhances our ability to discuss and improve our work and the work of our peers, both before and after publication. 

Much of the ecological science community shares one or both of these goals: to do the best possible science, and to do it in a way that is most useful and accessible to colleagues and to society at large. The goal of this year’s ecological unconference is to introduce as many people as possible to resources - both tools and people - that can help all of us achieve those goals, on our own, or together.

One way we as ecologists can quickly make our research more reproducible is the way we write. By simply using tools that make reproducing what we have done easy to do, we can avoid retracted papers, failed drugs, and ruined careers. 

*********

### What are you proposing?
We originally thought about a separate event from ESA itself, modeled after [Science Online](http://scienceonline.com/), incorporating a variety of topics. However, we thought testing the waters for this sort of non-traditional unconference format would be better in 2013. We are gathering ideas from the community (see “How do I make my voice heard?” below). The ideas on the wiki that get the most traction, and have 1-2 champions that are willing to see the idea through and lead the workshop at ESA will be turned in to proposals for ESA workshops. In addition, we will be submitting a proposal for an Ignite session at ESA. To summarise, we will be running:

+ A few workshops (at lunch hours, and half-day). Topics may include:
	+ Data sharing
	+ Data visualization
	+ Data management
	+ Alternatives to academia
	+ Blogging
	+ Social media
	+ Reproducible science writing
+ One Ignite session on “Tools for better/faster science”. See more about Ignite sessions [here](http://www.esa.org/minneapolis/ignite.php).
+ A “tweetup” event to socialize in a more relaxed atmosphere

These will all be loosely aggregated under the [#AltEcology](https://twitter.com/search/realtime?q=%23AltEcology) hashtag.

*********

### How do I make my voice heard?
We have set up a wiki in which anyone can contribute. Please share your ideas and voice your support for existing ones at the wiki [here](http://ecologyunconference.wikispaces.com/).  You can just throw ideas out there, or even propose new workshops and nominate people to lead them. We’re currently moving to transform existing ideas into ESA workshop and Ignite proposals to meet the [November 29 deadline](http://www.esa.org/minneapolis/workshop.php), but we’ll be incorporating input from the wiki right up to the meeting itself in August 2013.

*********

### Get in touch
If you have any questions/comments, let us know in the comments section below, tweet us (Sandra: [@sandramchung](https://twitter.com/sandramchung), Scott: [@recology_](https://twitter.com/recology_)), or email ([Sandra](mailto:sandra.m.chung@gmail.com), [Scott](mailto:myrmecocystus@gmail.com)). 

*********
#### Get the .md file used to create this post [at my github account](https://github.com/sckott/sckott.github.com/tree/master/_posts/2012-11-15-altecology.md).  Written in [Markdown](http://daringfireball.net/projects/markdown/), with help from [knitr](http://yihui.name/knitr/).

  </div>
  
  <div class="post">
    <h1>
      <a href="/2012/10/r2g2-package/">
        Displaying Your Data in Google Earth Using R2G2
      </a>
    </h1>

    <span class="post-date">24 Oct 2012</span>

    Have you ever wanted to easily visualize your ecology data in [Google Earth][googleearth]?  [R2G2][r2g2] is a new package for R, available via [R CRAN][rcran] and formally described in [this Molecular Ecology Resources article][MERarticle], which provides a user-friendly bridge between R and the Google Earth interface.  Here, we will provide a brief introduction to the package, including a short tutorial, and then encourage you to try it out with your own data!

[Nils Arrigo][nils], with some help from [Loren Albert][loren], [Mike Barker][mike], and Pascal Mickelson (one of the contributors to [Recology][recology]), has created a set of R tools to generate KML files to view data with geographic components.  Instead of just telling you what the tools can do, though, we will show you a couple of examples using publically available data.  Note: a number of individual files are linked to throughout the tutorial below, but just in case you would rather download all the tutorial files in one go, have at it ([tutorial zip file][tutorialfile]).

Among the basic tools in [R2G2][r2g2] is the ability to place features—like dots, shapes, or images (including plots you produced in R)— that represent discrete observations at specific geographical locations.  For example, in the figure below, we show the migratory path of a particular turkey vulture in autumn of three successive years (red = 2009; blue = 2010; green = 2011).  


<div align="center">
<img src="/public/img/R2G2tutorial/Vulture_Path.jpg" alt="Google Earth image with three successive years of a particular turkey vulture's migration" width="800"><br>
<small>Google Earth imagery showing migratory path of a particular turkey vulture in 2009, 2010, and 2011.</small>
</div>  


We use the *PolyLines2GE* function that is part of [R2G2][r2g2] to create line segments between the geographical coordinates which have been obtained from a turkey vulture tagged with a transponder (data accessed via the [Movebank Data Repository][movebank] and is from the [Turkey Vulture Acopian Center USA GPS][turkeyvulturestudy]).  The *PolyLines2GE* function looks like the following:  

{% highlight r %}
PolyLines2GE(coords = vulture_path[,2:3],  
			nesting = vulture_path[,1],  
			colors = "auto",  
			goo = "Vulture_Path.kml",  
			maxAlt = 1e4,  
			fill = FALSE,  
			closepoly = FALSE,  
			lwd = 2,  
			extrude = 0)
{% endhighlight %}  

It expects to receive an array ("coords") containing latitude and longitude coordinates in decimal degrees. Additionally, each individual coordinate has a flag associated with it ("nesting") so that each data series can be distinguished.  Illustrating what you need is easier than explaining:
{% highlight text %}
nesting longitude latitude
1	long1A		lat1A
1	long1B		lat1B
1	long1C		lat1C
2	long2A		lat2A
2	long2B		lat2B
3	long3A		lat3A
3	long3B		lat3B
3	long3C		lat3C
{% endhighlight %}
Feeding the columns of this array to the function results in three differently colored lines:  the first would connect the coordinates 1A-1B-1C, while the second would connect 2A-2B, and the third would connect 3A-3B-3C.  The only other user-defined input that is strictly necessary is the output file name ("Vulture_Path.kml" in this case).  The other options—which allow you control of the appearance of the lines and of the altitude at which your line displays in Google Earth—have reasonable defaults that are well-documented in the function definition itself.  Check out this example in Google Earth [by downloading the KML file][vultureKML].  Alternatively, [download the annotated R script][vultureR] and generate the KML file for yourself.

Now, let's say you wanted to get a sense of the range and abundance of two congeneric species.  In this second example, we use the *Hist2GE* function to create a histogram—overlaid on the surface of the earth—which shows the species distribution of *Mimulus lewisii* (red) and *Mimulus nasutus* (blue) in North America.  


<div align="center">
<img src="/public/img/R2G2tutorial/Mimulus_Distribution.jpg" alt="Google Earth image showing the distribution of Mimulus in North America" width="800"><br>
<small>Google Earth imagery showing the species distribution of <em>Mimulus lewisii</em> and <em>Mimulus nasutus</em></small>
</div>  


As you might expect, each polygon represents an occurrence of the species in question, while the height of the polygon represents the abundance of the species at that geographic location.  Species occurring within a particular distance of each other have been grouped together for the histogram.  For this example, we retrieve data from the [GBIF][gbif] database from within R (see the example code for how that is done).  Inputs to the Hist2GE function are:
{% highlight r %}
Hist2GE(coords = MyCompleteData[, 8:7],  
		species = MyCompleteData[, 1],  
		grid = grid10000,  
		goo = "Mimulus",  
		nedges = 6,  
		orient = 45,  
		maxAlt = 1e4)
{% endhighlight %}
As in the first example, the function expects to receive an array containing the longitude and latitude ("coords"), a vector distinguishing individual observations ("species"), and an output file name ("goo").  In this case, however, we also need to specify the size of the grid we will use to group observations together to construct the histogram.  Several pre-defined grid sizes are included in the package to do this grouping; these all cover large geographic areas and therefore must account for the curvature of the earth.  Here is a list of these pre-defined grids:

<table border="1">
    <tr><th>Grid Name</th><th>Approximate Area of Grid Division </th></tr>
    <tr><td>grid20000</td><td>25,500 sq. km</td></tr>
    <tr><td>grid10000</td><td>51,000 sq. km</td></tr>
    <tr><td>grid5000</td><td>102,000 sq. km</td></tr>
    <tr><td>grid500</td><td>1,020,000 sq. km</td></tr>
    <tr><td>grid50</td><td>10,200,000 sq. km</td></tr>
</table>  

For smaller geographic areas (less than 25,000 square kilometers, or an area of about 158 km per side), you can customize the grid size by specifying the bounds of your region of interest in decimal degrees, as well as the coarseness of the grid within that region.  While it is possible to use this custom grid definition for larger sizes, beware that not all areas defined thusly will be of equal size due to the earth's curvature (obviously the bigger you go, the worse it gets...).  Finally, you again have control over the display parameters of the histogram.  In particular, the maximum altitude ("maxAlt") controls how high the tallest bar in the histogram will go.  Here is [the resulting KML file][mimulusKML], as well as [the annotated R script][mimulusR] so you can further explore the example.

More complex visual representations are also possible using [R2G2][r2g2].  For instance, you can also create contour plots or phylogenies overlaid directly on the surface of the earth.  We included a couple examples of this type in [our Molecular Ecology Resources article][MERarticle], and if the response seems good, we may post a follow up tutorial showing how we went about creating those examples.

It is our sincere hope that you will use the tools in [R2G2][r2g2] to more effectively visualize the geographical aspects of your data.  In particular, we are excited about the potential for incorporating [R2G2][r2g2] into data analysis pipelines connecting analysis in R with data visualization and exploration in Google Earth.  Ultimately, the inclusion of KML files as supplementary materials to journal articles should also enrich one's understanding of the data being presented!


<big><b>Note:  If you make something cool using <a href="http://cran.r-project.org/web/packages/R2G2/index.html">R2G2</a>, please post a link to your KML file in the comments; we would love to see!</b></big>


Citation information for [R2G2][r2g2]:  
*Arrigo, N., Albert, L. P., Mickelson, P. G. and Barker, M. S. (2012), Quantitative visualization of biological data in Google Earth using R2G2, an R CRAN package. Molecular Ecology Resources. doi: 10.1111/1755-0998.12012*

[googleearth]: http://earth.google.com
[rcran]: http://cran.r-project.org/
[MERarticle]: http://onlinelibrary.wiley.com/doi/10.1111/1755-0998.12012/abstract
[nils]: http://barkerlab.net/nils.html
[loren]: http://portal.environment.arizona.edu/students/profiles/loren-albert
[mike]: http://barkerlab.net/mike.html
[recology]: http://sckott.github.io/about.html
[gbif]: http://data.gbif.org/
[r2g2]: http://cran.r-project.org/web/packages/R2G2/index.html
[vultureKML]: /public/img/R2G2tutorial/Vulture_Path.kml
[vultureR]: /public/img/R2G2tutorial/Vulture_Path.R
[movebank]: http://movebank.org/
[turkeyvulturestudy]: http://movebank.org/movebank/#page%3Dstudies%2Cpath%3Dstudy481458
[mimulusKML]: /public/img/R2G2tutorial/Mimulus_Distribution.kml
[mimulusR]: /public/img/R2G2tutorial/Mimulus_Distribution.R
[tutorialfile]: /public/img/R2G2tutorial/R2G2tutorial.zip

  </div>
  
</div>

<!-- Pagination links -->
<div class="pagination">
  
    <a href="/page27" class="older">Older</a>
  
  
    
      <a href="/page25" class="newer">Newer</a>
    
  
</div>

    </div>

    <!-- for bootstrap tooltips -->
    <script type="text/javascript">
      $("[data-toggle=\"tooltip\"]").tooltip();
    </script>

  </body>

  <footer>
  <!-- Disqus code -->
  <script type="text/javascript">
      /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
      var disqus_shortname = 'recology'; // required: replace example with your forum shortname

      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function () {
          var s = document.createElement('script'); s.async = true;
          s.type = 'text/javascript';
          s.src = 'http://' + disqus_shortname + '.disqus.com/count.js';
          (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
      }());
  </script>

  <!-- google analytics -->
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-63197374-1', 'auto');
    ga('send', 'pageview');
  </script>
</footer>

</html>
