<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <title>
    
    Recology, R ecology
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:300,400italic,400,600,700|Abril+Fatface">
  

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/touch-icon-144-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.png">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body class="theme-base-0f layout-reverse">

    <header class="masthead">
      <div class="masthead-inner">
        <h1>Recology</h1>
        <!-- <h1> <a href="http://recology.info/">Recology</a></h1> -->
        <p class="lead">R & science</p>

        <div class="colophon">
          <ul class="colophon-links">
            <li>
              <a href="/"><i class="fa fa-home fa-lg"></i></a>&nbsp;
              <a href="/about"><i class="fa fa-info-circle fa-lg"></i></a>&nbsp;
              <a href="/rresources"><i class="fa fa-book fa-lg"></i></a>&nbsp;
              <a href="http://feeds.feedburner.com/github/wpna"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
              <a href="https://twitter.com/recology_"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
              <a href="https://alpha.app.net/intent/follow/?user_id=@recology_"><i class="fa fa-font fa-lg"></i></a>
              <a href="fork"><i class="fa fa-spinner fa-lg"></i></a>
            </li>
          </ul>
          <small><a href="https://github.com/mdo/hyde">Hyde</a> from <a href="https://twitter.com/mdo" target="_blank">@mdo</a>.</small>
        </div>
      </div>
    </header>

    <div class="content container">
      <div class="posts">
  
  <div class="post">
    <h1>
      <a href="/2013/06/17/couch/">
        Stashing and playing with raw data locally from the web
      </a>
    </h1>

    <span class="post-date">17 Jun 2013</span>

    <p>It is getting easier to get data directly into R from the web. Often R packages that retrieve data from the web return useful R data structures to users like a data.frame. This is a good thing of course to make things user friendly.</p>

<p>However, what if you want to drill down into the data that's returned from a query to a database in R?  What if you want to get that nice data.frame in R, but you think you may want to look at the raw data later? The raw data from web queries are often JSON or XML data. This type of data, especially JSON, can be easily stored in schemaless so-called NoSQL databases, and queried later.</p>

<p>A brief aside: What are JSON and XML? This is what JSON looks like (ps if you ever wonder if your JSON is correct, go <a href="http://jsonlint.com/">here</a>):</p>

<div class="highlight"><pre><code class="bash"><span class="o">{</span>
  <span class="s2">&quot;name&quot;</span>: <span class="s2">&quot;joe&quot;</span>,
  <span class="s2">&quot;hobby&quot;</span>: <span class="s2">&quot;codemonkey&quot;</span>,
  <span class="s2">&quot;lives&quot;</span>: <span class="o">[</span>
      <span class="o">{</span>
          <span class="s2">&quot;city&quot;</span>: <span class="s2">&quot;San Jose&quot;</span>,
          <span class="s2">&quot;state&quot;</span>: <span class="s2">&quot;CA&quot;</span>
      <span class="o">}</span>
  <span class="o">]</span>
<span class="o">}</span>
</code></pre></div>


<p>This is what XML looks like:</p>

<div class="highlight"><pre><code class="bash">&lt;?xml <span class="nv">version</span><span class="o">=</span><span class="s2">&quot;1.0&quot;</span> <span class="nv">encoding</span><span class="o">=</span><span class="s2">&quot;UTF-8&quot;</span> ?&gt;
  &lt;name&gt;joe&lt;/name&gt;
    &lt;hobby&gt;codemonkey&lt;/hobby&gt;
    &lt;lives&gt;
        &lt;city&gt;San Jose&lt;/city&gt;
        &lt;state&gt;CA&lt;/state&gt;
    &lt;/lives&gt;
</code></pre></div>


<p>But don't worry if it looks complicated - the project I talk about below, sofa, tries to make the interface to JSON and XML easy. Web APIs almost always return either JSON or XML, so this is the raw data.</p>

<p>So here's the use case I imagine, or workflow:</p>

<ul>
<li>Query a database on the web, and choose to write the raw data to a local database.</li>
<li>Do whatever you want with the output R object - analyze, visualize, etc.</li>
<li>Now you want to go back and search through some of the raw data. But, that query took an hour. Since you wrote it to a local database, you can search the data.</li>
<li>If you hadn't written it locally, you would have to make a new web call.</li>
</ul>


<p>Note that if you are doing calls to web APIs that get small amounts of data you don't need to worry too much as you can easily just do the call again.</p>

<p>I've started an R package to interact with the NoSQL database <a href="http://couchdb.apache.org/">CouchDB</a>. CouchDB is a schemaless database that speaks JSON, so you can store JSON and get back JSON (don't worry XML, we got you covered - we can just wrap the XML in JSON before putting it in CouchDB). What's especially cool is you can interact with CouchDB via <a href="http://docs.couchdb.org/en/latest/api-basics.html">a RESTful API</a>. CouchDB doesn't have full text search built in (though you can build map-reduce <em>Views</em>, basically preset queries on the database), so I added functions (and docs to help) to interact with the <a href="https://github.com/elasticsearch/elasticsearch-river-couchdb/blob/master/README.md">CouchDB River plugin</a> for <a href="http://www.elasticsearch.org/">Elasticsearch</a>, which provides powerful full text search via an API interface. But nevermind the tech details - all this just means you can search on the full text of your stored data.</p>

<p>There are plenty of databases you can interact with from R, so why CouchDB? For one, it makes a lot of sense to write to a NoSQL database since this blog post is dealing with a use case writing JSON, which isn't a good fit for databases like MySQL, SQLite, PostgreSQL, etc. (<a href="http://wiki.postgresql.org/wiki/What's_new_in_PostgreSQL_9.2#JSON_datatype">though postgres allows you to write JSON</a>). It didn't have to be CouchDB, but at least to me it seems relatively easy to install, you can interact with it via an HTTP API (if you're into that, which I am), and it has a nice web interface (navigate to <a href="http://localhost:5984/_utils/">http://localhost:5984/_utils/</a> after starting <code>couchdb</code>).</p>

<p>Is this for the casual R user? Probably not. But, I imagine there are R users out there that want some more flexibility when it comes to interacting with web data in R. It is nice and tidy to get back an R data.frame from a web call, but having the raw data at your fingertips could be super powerful. I'll describe using an R package to interact with a web database with <code>sofa</code> baked in, and discuss a bit about the functions within <code>sofa</code>.</p>

<hr />

<h3>First start CouchDB in your terminal</h3>

<p>You can do this from anywhere in your directory. See <a href="http://couchdb.apache.org/">here</a> for instructions on how to install CouchDB.</p>

<pre><code class="bash">couchdb
</code></pre>

<h3>Then start elasticsearch in your terminal</h3>

<p>See <a href="https://github.com/sckott/sofa">here</a> for instructions on how to install Elasticsearch and the River CouchDB plugin.</p>

<pre><code class="bash">cd /usr/local/elasticsearch
bin/elasticsearch -f
</code></pre>

<hr />

<h3>Install sofa</h3>

<div class="highlight"><pre><code class="r"><span class="c1"># Uncomment these lines if you don&#39;t have these packages installed</span>
<span class="c1"># install.packages(&#39;devtools&#39;) library(devtools) install_github(&#39;sofa&#39;,</span>
<span class="c1"># &#39;schamberlain&#39;) install_github(&#39;alm&#39;, &#39;ropensci&#39;, ref=&#39;couch&#39;)</span>
library<span class="p">(</span>sofa<span class="p">)</span>
library<span class="p">(</span>alm<span class="p">)</span>
</code></pre></div>


<hr />

<h3>Simultaneously write data to CouchDB along with API calls using the alm package to get altmetrics data on PLoS papers. Ping to make sure CouchDB is on</h3>

<div class="highlight"><pre><code class="r">sofa_ping<span class="p">()</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">  couchdb   version 
&quot;Welcome&quot;   &quot;1.2.1&quot; 
</code></pre></div>


<hr />

<h3>Create a new database</h3>

<div class="highlight"><pre><code class="r">sofa_createdb<span class="p">(</span>dbname <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">  ok 
TRUE 
</code></pre></div>


<hr />

<h3>Write couchdb database name to options</h3>

<div class="highlight"><pre><code class="r">options<span class="p">(</span>couch_db_name <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">)</span>
</code></pre></div>


<hr />

<h3>List the databases</h3>

<div class="highlight"><pre><code class="r">sofa_listdbs<span class="p">()</span>
</code></pre></div>




<div class="highlight"><pre><code class="text"> [1] &quot;_replicator&quot;                &quot;_users&quot;                    
 [3] &quot;alm_couchdb&quot;                &quot;alm_db&quot;                    
 [5] &quot;dudedb&quot;                     &quot;example&quot;                   
 [7] &quot;poop&quot;                       &quot;rplos_db&quot;                  
 [9] &quot;shit&quot;                       &quot;shitty&quot;                    
[11] &quot;shitty2&quot;                    &quot;test_suite_db&quot;             
[13] &quot;test_suite_db/with_slashes&quot; &quot;test_suite_reports&quot;        
[15] &quot;testr2couch&quot;                &quot;twitter_db&quot;                
</code></pre></div>


<hr />

<h3>Search for altmetrics normally, w/o writing to a database</h3>

<div class="highlight"><pre><code class="r">head<span class="p">(</span>alm<span class="p">(</span>doi <span class="o">=</span> <span class="s">&quot;10.1371/journal.pone.0029797&quot;</span><span class="p">))</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">          .id pdf html shares groups comments likes citations total
1   bloglines  NA   NA     NA     NA       NA    NA         0     0
2   citeulike  NA   NA      1     NA       NA    NA        NA     1
3    connotea  NA   NA     NA     NA       NA    NA         0     0
4    crossref  NA   NA     NA     NA       NA    NA         6     6
5      nature  NA   NA     NA     NA       NA    NA         4     4
6 postgenomic  NA   NA     NA     NA       NA    NA         0     0
</code></pre></div>


<hr />

<h3>Search for altmetrics normally, while writing to a database</h3>

<div class="highlight"><pre><code class="r">head<span class="p">(</span>alm<span class="p">(</span>doi <span class="o">=</span> <span class="s">&quot;10.1371/journal.pone.0029797&quot;</span><span class="p">,</span> write2couch <span class="o">=</span> <span class="kc">TRUE</span><span class="p">))</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">          .id pdf html shares groups comments likes citations total
1   bloglines  NA   NA     NA     NA       NA    NA         0     0
2   citeulike  NA   NA      1     NA       NA    NA        NA     1
3    connotea  NA   NA     NA     NA       NA    NA         0     0
4    crossref  NA   NA     NA     NA       NA    NA         6     6
5      nature  NA   NA     NA     NA       NA    NA         4     4
6 postgenomic  NA   NA     NA     NA       NA    NA         0     0
</code></pre></div>


<hr />

<h3>Make lots of calls, and write them simultaneously</h3>

<div class="highlight"><pre><code class="r"><span class="c1"># install_github(&#39;rplos&#39;, &#39;ropensci&#39;)</span>
library<span class="p">(</span>rplos<span class="p">)</span>
dois <span class="o">&lt;-</span> searchplos<span class="p">(</span>terms <span class="o">=</span> <span class="s">&quot;evolution&quot;</span><span class="p">,</span> fields <span class="o">=</span> <span class="s">&quot;id&quot;</span><span class="p">,</span> limit <span class="o">=</span> <span class="m">100</span><span class="p">)</span>
out <span class="o">&lt;-</span> alm<span class="p">(</span>doi <span class="o">=</span> as.character<span class="p">(</span>dois<span class="p">[,</span> <span class="m">1</span><span class="p">]),</span> write2couch <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
lapply<span class="p">(</span>out<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span> head<span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">$`01`
          .id pdf html shares groups comments likes citations total
1   bloglines  NA   NA     NA     NA       NA    NA         0     0
2   citeulike  NA   NA      0     NA       NA    NA        NA     0
3    connotea  NA   NA     NA     NA       NA    NA         0     0
4    crossref  NA   NA     NA     NA       NA    NA         0     0
5      nature  NA   NA     NA     NA       NA    NA         0     0
6 postgenomic  NA   NA     NA     NA       NA    NA         0     0

$`02`
          .id pdf html shares groups comments likes citations total
1   bloglines  NA   NA     NA     NA       NA    NA         0     0
2   citeulike  NA   NA      1     NA       NA    NA        NA     1
3    connotea  NA   NA     NA     NA       NA    NA         0     0
4    crossref  NA   NA     NA     NA       NA    NA         2     2
5      nature  NA   NA     NA     NA       NA    NA         0     0
6 postgenomic  NA   NA     NA     NA       NA    NA         0     0
</code></pre></div>


<hr />

<h3>Writing data to CouchDB does take a bit longer</h3>

<div class="highlight"><pre><code class="r">system.time<span class="p">(</span>alm<span class="p">(</span>doi <span class="o">=</span> as.character<span class="p">(</span>dois<span class="p">[,</span> <span class="m">1</span><span class="p">])[</span><span class="m">1</span><span class="o">:</span><span class="m">60</span><span class="p">],</span> write2couch <span class="o">=</span> <span class="kc">FALSE</span><span class="p">))</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">   user  system elapsed 
  1.739   0.016   4.554 
</code></pre></div>




<div class="highlight"><pre><code class="r">system.time<span class="p">(</span>alm<span class="p">(</span>doi <span class="o">=</span> as.character<span class="p">(</span>dois<span class="p">[,</span> <span class="m">1</span><span class="p">])[</span><span class="m">1</span><span class="o">:</span><span class="m">60</span><span class="p">],</span> write2couch <span class="o">=</span> <span class="kc">TRUE</span><span class="p">))</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">   user  system elapsed 
  3.579   0.062   6.460 
</code></pre></div>


<hr />

<h3>Search using elasticsearch</h3>

<h4>tell elasticsearch to start indexing your database</h4>

<div class="highlight"><pre><code class="r">elastic_start<span class="p">(</span>dbname <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">$ok
[1] TRUE
</code></pre></div>


<hr />

<h4>Search your database</h4>

<div class="highlight"><pre><code class="r">out <span class="o">&lt;-</span> elastic_search<span class="p">(</span>dbname <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">,</span> q <span class="o">=</span> <span class="s">&quot;twitter&quot;</span><span class="p">,</span> parse_ <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
out<span class="o">$</span>hits<span class="o">$</span>total
</code></pre></div>




<div class="highlight"><pre><code class="text">[1] 679
</code></pre></div>


<hr />

<h3>Using views</h3>

<h4>Write a view - here letting key be the default of null</h4>

<div class="highlight"><pre><code class="r">sofa_view_put<span class="p">(</span>dbname <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">,</span> design_name <span class="o">=</span> <span class="s">&quot;myview&quot;</span><span class="p">,</span> value <span class="o">=</span> <span class="s">&quot;doc.baseurl&quot;</span><span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">$ok
[1] TRUE

$id
[1] &quot;_design/myview&quot;

$rev
[1] &quot;1-e7c17cff1b96e4595c3781da53e16ad8&quot;
</code></pre></div>


<hr />

<h4>Get info on your new view</h4>

<div class="highlight"><pre><code class="r">sofa_view_get<span class="p">(</span>dbname <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">,</span> design_name <span class="o">=</span> <span class="s">&quot;myview&quot;</span><span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">$`_id`
[1] &quot;_design/myview&quot;

$`_rev`
[1] &quot;1-e7c17cff1b96e4595c3781da53e16ad8&quot;

$views
$views$foo
                                    map 
&quot;function(doc){emit(null,doc.baseurl)}&quot; 
</code></pre></div>


<hr />

<h4>Get data using a view</h4>

<div class="highlight"><pre><code class="r">out <span class="o">&lt;-</span> sofa_view_search<span class="p">(</span>dbname <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">,</span> design_name <span class="o">=</span> <span class="s">&quot;myview&quot;</span><span class="p">)</span>
length<span class="p">(</span>out<span class="o">$</span>rows<span class="p">)</span>  <span class="c1"># 160 results</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">[1] 161
</code></pre></div>




<div class="highlight"><pre><code class="r">sapply<span class="p">(</span>out<span class="o">$</span>rows<span class="p">,</span> <span class="kr">function</span><span class="p">(</span>x<span class="p">)</span> x<span class="o">$</span>value<span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>  <span class="c1"># the values, just the API call URLs</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">[1] &quot;http://alm.plos.org/api/v3/articles&quot;
[2] &quot;http://alm.plos.org/api/v3/articles&quot;
[3] &quot;http://alm.plos.org/api/v3/articles&quot;
[4] &quot;http://alm.plos.org/api/v3/articles&quot;
[5] &quot;http://alm.plos.org/api/v3/articles&quot;
</code></pre></div>


<hr />

<h4>Delete the view</h4>

<div class="highlight"><pre><code class="r">sofa_view_del<span class="p">(</span>dbname <span class="o">=</span> <span class="s">&quot;alm_db&quot;</span><span class="p">,</span> design_name <span class="o">=</span> <span class="s">&quot;myview&quot;</span><span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">[1] &quot;&quot;
</code></pre></div>


<hr />

<h2>What now?</h2>

<p>Well, if no one uses this, then probably nothing. Though, if people think this could be useful:</p>

<ul>
<li>It would be cool to make easy hooks into any package making web calls to allow users to write data to CouchDB if they choose to, sort of like the example above with rplos.</li>
<li>Perhaps automate some of the setup for CouchDB for users, making system calls so they don't have to.</li>
<li>Performance: As shown above, simultaneously writing data to CouchDB takes longer than not doing so - removing this time difference will make writing to couch more palatable.</li>
</ul>


<h2>What do you think?</h2>

<p>What is your reaction to this post?  Do you have a need for this sort of tool?  Do you have similar use cases that could be addressed with <code>sofa</code>?</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2013/06/01/fylopic/">
        Fylopic, an R wrapper to Phylopic
      </a>
    </h1>

    <span class="post-date">01 Jun 2013</span>

    <h2>What is PhyloPic?</h2>

<p>PhyloPic is an awesome new service - I'll let the creator, <a href="http://tmkeesey.net/">Mike Keesey</a>, explain what it is (paraphrasing here):</p>

<blockquote><p>PhyloPic stores silhouette images of organisms, and each image is associated with taxonomic names, and stores the taxonomy of all taxa, allowing searching by taxonomic names. Anyone can submit silhouettes to PhyloPic.</p></blockquote>

<p>What is a silhouette?  It's like this:</p>

<p><img src="http://phylopic.org/assets/images/submissions/bedd622a-4de2-4067-8c70-4aa44326d229.128.png" alt="A silhouette" /></p>

<p><em>by Gareth Monger</em></p>

<p>What makes PhyloPic not just awesome, but super awesome? All or most images are licensed under <a href="http://creativecommons.org/">Creative Commons licenses</a>. This means you can use the silhouettes without having to ask or pay - just attribute.</p>

<hr />

<h2>What is fylopic?</h2>

<p>The idea behind Fylopic is to create modular bits and pieces (i.e., functions) to allow you to add silhouettes to not only ggplot2 plots, but base plots as well. That is, you can simply load fylopic in your R session, and add some silhouettes to your phylogeny, or your barchart, etc. - that is, <code>fylopic</code> is meant to be a helper in your workflow to add in silhouettes to visualizations.</p>

<p>Some people prefer base plots while others prefer ggplot2 plots (me!), so it would be nice to have both options. Phylogenies at the moment render faster in base plots. I don't yet have implementations for base plots, but will come soon, or you can send a pull request to add it.</p>

<p>One interesting use case could be to be able to get a set of silhouettes, then get a phylogeny for taxa associatd with the silhouettess using the NCBI taxonomy, but it's not easily available yet (though I may be able to use <a href="https://github.com/bendmorris/phylocommons">Ben Morris' phylocommons</a> soon. This isn't doable yet, so in the example below the function <code>make_phylo</code> creates a phylogeny using <code>ape::rcoal</code>.</p>

<p>You could also do the reverse -> you have a phylogeny and then you could search Phylopic for silhouettes.</p>

<hr />

<h2>Info</h2>

<p>Check out the Phylopic website <a href="http://phylopic.org/">here</a>, and Phylopic API developer documentation <a href="http://phylopic.org/api/">here</a>.</p>

<p>Also check out Ben Morris' Python wrapper to Phylopic <a href="https://github.com/bendmorris/python-phylopic">here</a>.</p>

<hr />

<h2>What can you do with fylopic?</h2>

<hr />

<h4>Install fylopic</h4>

<div class="highlight"><pre><code class="r">install.packages<span class="p">(</span><span class="s">&quot;devtools&quot;</span><span class="p">)</span>
library<span class="p">(</span>devtools<span class="p">)</span>
install_github<span class="p">(</span><span class="s">&quot;fylopic&quot;</span><span class="p">,</span> <span class="s">&quot;sckott&quot;</span><span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="r">library<span class="p">(</span>fylopic<span class="p">,</span> quietly <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>


<hr />

<h4>Plot a phylogeny with silhouettes at the tips</h4>

<p>Here, I search for names based on keyword <em>Homo sapiens</em> - which returns many matche codes. With those results we search for any silhouettes associated with those codes. Then we download images. Finally, make a phylogeny with the silhouettes at the tips. Note that in this eample the phylogeny is just a random coalescent tree made using <code>ape::rcoal</code> - obviously, in the real world you'd want to do something more useful.</p>

<div class="highlight"><pre><code class="r"><span class="c1">## search on Homo sapiens</span>
searchres <span class="o">&lt;-</span> search_text<span class="p">(</span>text <span class="o">=</span> <span class="s">&quot;Homo sapiens&quot;</span><span class="p">,</span> options <span class="o">=</span> <span class="s">&quot;names&quot;</span><span class="p">)</span>

<span class="c1">### which returns UUIDs</span>
searchres<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">]</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">[1] &quot;74aea16b-666b-497a-b2cb-72201ad75a8e&quot;
[2] &quot;1ee65cf3-53db-4a52-9960-a9f7093d845d&quot;
[3] &quot;cc9ad8ee-3a82-4add-8d50-bc78f4ff6956&quot;
</code></pre></div>




<div class="highlight"><pre><code class="r"><span class="c1">## search for images based on the UUIds</span>
output <span class="o">&lt;-</span> search_images<span class="p">(</span>uuid <span class="o">=</span> searchres<span class="p">,</span> options <span class="o">=</span> c<span class="p">(</span><span class="s">&quot;pngFiles&quot;</span><span class="p">,</span> <span class="s">&quot;credit&quot;</span><span class="p">,</span> 
    <span class="s">&quot;canonicalName&quot;</span><span class="p">))</span>

<span class="c1">### we got eight matches</span>
output
</code></pre></div>




<div class="highlight"><pre><code class="text">$`15444b9c-f17f-4d6e-89b5-5990096bcfb0`
$`15444b9c-f17f-4d6e-89b5-5990096bcfb0`$supertaxa
[1] &quot;e547cd01-7dd1-495b-8239-52cf9971a609&quot;
[2] &quot;bd88f674-6976-4cb2-a46e-e6a12a8ba463&quot;


$`fedf0e5f-f20a-442c-accf-eb84a3af8c6b`
$`fedf0e5f-f20a-442c-accf-eb84a3af8c6b`$supertaxa
[1] &quot;e547cd01-7dd1-495b-8239-52cf9971a609&quot;
[2] &quot;bd88f674-6976-4cb2-a46e-e6a12a8ba463&quot;


$`a88d3a4c-44d3-409e-87b6-516bd188c709`
$`a88d3a4c-44d3-409e-87b6-516bd188c709`$supertaxa
[1] &quot;e547cd01-7dd1-495b-8239-52cf9971a609&quot;
[2] &quot;bd88f674-6976-4cb2-a46e-e6a12a8ba463&quot;


$`d88164ec-3152-444b-b41c-4757a344a764`
$`d88164ec-3152-444b-b41c-4757a344a764`$supertaxa
[1] &quot;9c6af553-390c-4bdd-baeb-6992cbc540b1&quot;


$`da5eaeb7-1ed2-4b2e-ad4a-49993881d706`
$`da5eaeb7-1ed2-4b2e-ad4a-49993881d706`$supertaxa
[1] &quot;9c6af553-390c-4bdd-baeb-6992cbc540b1&quot;
</code></pre></div>




<div class="highlight"><pre><code class="r"><span class="c1">## download images</span>
myobjs <span class="o">&lt;-</span> get_image<span class="p">(</span>uuids <span class="o">=</span> output<span class="p">,</span> size <span class="o">=</span> <span class="s">&quot;128&quot;</span><span class="p">)</span>

<span class="c1">## make the phylogeny</span>
make_phylo<span class="p">(</span>pngobj <span class="o">=</span> myobjs<span class="p">)</span>
</code></pre></div>


<p><img src="/public/img/2013-06-01-fylopic/unnamed-chunk-1.png" alt="center" /></p>

<hr />

<h4>Plot a silhouette behind a plot</h4>

<p>Notice in the below example that you can use normal <code>ggplot2</code> syntax, and simply add another layer (<code>add_phylopic</code> from <code>fylopic</code>) to the plot.</p>

<div class="highlight"><pre><code class="r">library<span class="p">(</span>ggplot2<span class="p">)</span>
img <span class="o">&lt;-</span> get_image<span class="p">(</span><span class="s">&quot;27356f15-3cf8-47e8-ab41-71c6260b2724&quot;</span><span class="p">,</span> size <span class="o">=</span> <span class="s">&quot;512&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]]</span>
qplot<span class="p">(</span>x <span class="o">=</span> Sepal.Length<span class="p">,</span> y <span class="o">=</span> Sepal.Width<span class="p">,</span> data <span class="o">=</span> iris<span class="p">,</span> geom <span class="o">=</span> <span class="s">&quot;point&quot;</span><span class="p">)</span> <span class="o">+</span> add_phylopic<span class="p">(</span>img<span class="p">)</span>
</code></pre></div>


<p><img src="/public/img/2013-06-01-fylopic/unnamed-chunk-2.png" alt="center" /></p>

<hr />

<h2>What's next?</h2>

<p>This is a side project, so if anyone has interest in helping please do contribute code, report bugs, request features, etc.</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2013/05/27/rbison/">
        BISON USGS species occurrence data
      </a>
    </h1>

    <span class="post-date">27 May 2013</span>

    <p>The USGS recently released a way to search for and get species occurrence records for the USA. The service is called <a href="http://bison.usgs.ornl.gov/">BISON</a> (Biodiversity Information Serving Our Nation). The service has <a href="http://bison.usgs.ornl.gov/">a web interface</a> for human interaction in a browser, and <a href="http://bison.usgs.ornl.gov/services.html">two APIs</a> (application programming interface) to allow machines to interact with their database. One of the APIs allows you to search and retrieve data, and the other gives back maps as either a heatmap or a species occurrence map. The latter is more appropriate for working in a browser, so I'll leave that to the web app folks.</p>

<p>The Core Science Analytics and Synthesis (CSAS) program of the US Geological Survey are responsible for BISON, and are the US node of the Global Biodiversity Information Facility (GBIF). BISON data is nested within that of GBIF, but has (or wil have?) additional data not in GBIF, as described on their <em>About</em> page:</p>

<blockquote><p>BISON has been initiated with the 110 million records GBIF makes available from the U.S. and is integrating millions more records from other sources each year</p></blockquote>

<p>Have a look at their <em>Data providers</em> and <em>Statistics</em> tabs on the BISON website, which list where data comes from and how many searches and downloads have been done on each data provider.</p>

<p>We (rOpenSci) started an R package to interact with the BISON search API >> <code>rbison</code>. You may be thinking, but if the data in BISON is also in GBIF, why both making another R package for BISON? Good question. As I just said, BISON will have some data GBIF won't have. Also, the services (search API and map service) are different than those of GBIF.</p>

<p>Check out the package on GitHub here <a href="https://github.com/ropensci/rbison">https://github.com/ropensci/rbison</a>.</p>

<p>Here is a quick run through of some things you can do with <code>rbison</code>.</p>

<hr />

<h3>Install ribson</h3>

<div class="highlight"><pre><code class="r"><span class="c1"># Install rbison from GitHub using devtools, uncomment to install</span>
<span class="c1"># install.packages(&#39;devtools&#39;) library(devtools) install_github(&#39;rbison&#39;,</span>
<span class="c1"># &#39;ropensci&#39;)</span>
library<span class="p">(</span>rbison<span class="p">)</span>
</code></pre></div>


<hr />

<h3>Search the BISON database for, of course, bison</h3>

<div class="highlight"><pre><code class="r"><span class="c1"># Do the search</span>
out <span class="o">&lt;-</span> bison<span class="p">(</span>species <span class="o">=</span> <span class="s">&quot;Bison bison&quot;</span><span class="p">,</span> type <span class="o">=</span> <span class="s">&quot;scientific_name&quot;</span><span class="p">,</span> start <span class="o">=</span> <span class="m">0</span><span class="p">,</span> count <span class="o">=</span> <span class="m">10</span><span class="p">)</span>

<span class="c1"># Check that the returned object is the right class (&#39;bison&#39;)</span>
class<span class="p">(</span>out<span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">[1] &quot;bison&quot;
</code></pre></div>


<h4>Get a summary of the data</h4>

<div class="highlight"><pre><code class="r">bison_data<span class="p">(</span>out<span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">  total observation fossil specimen unknown
1   761          30      4      709      18
</code></pre></div>


<h4>Summary by counties (just the first 6 rows)</h4>

<div class="highlight"><pre><code class="r">head<span class="p">(</span>bison_data<span class="p">(</span>input <span class="o">=</span> out<span class="p">,</span> datatype <span class="o">=</span> <span class="s">&quot;counties&quot;</span><span class="p">))</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">  record_id total county_name      state
1     48295     7    Lipscomb      Texas
2     41025    15      Harney     Oregon
3     49017     8    Garfield       Utah
4     35031     2    McKinley New Mexico
5     56013     1     Fremont    Wyoming
6     40045     2       Ellis   Oklahoma
</code></pre></div>


<h4>Summary of states</h4>

<div class="highlight"><pre><code class="r">bison_data<span class="p">(</span>input <span class="o">=</span> out<span class="p">,</span> datatype <span class="o">=</span> <span class="s">&quot;states&quot;</span><span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">      record_id total county_fips
1    Washington     1          53
2         Texas     8          48
3    New Mexico     8          35
4          Iowa     1          19
5       Montana     9          30
6       Wyoming   155          56
7        Oregon    15          41
8      Oklahoma    14          40
9        Kansas    10          20
10      Arizona     1          04
11       Alaska    29          02
12         Utah    16          49
13     Colorado    17          08
14     Nebraska     1          31
15 South Dakota    61          46
</code></pre></div>


<hr />

<h3>Map the results</h3>

<div class="highlight"><pre><code class="r"><span class="c1"># Search for Ursus americanus (american black bear)</span>
out <span class="o">&lt;-</span> bison<span class="p">(</span>species <span class="o">=</span> <span class="s">&quot;Ursus americanus&quot;</span><span class="p">,</span> type <span class="o">=</span> <span class="s">&quot;scientific_name&quot;</span><span class="p">,</span> start <span class="o">=</span> <span class="m">0</span><span class="p">,</span> 
    count <span class="o">=</span> <span class="m">200</span><span class="p">)</span>

<span class="c1"># Sweet, got some data</span>
bison_data<span class="p">(</span>out<span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">  total observation fossil specimen literature unknown centroid
1  3792          59    125     3522         47      39       78
</code></pre></div>


<h3>Make some maps! Note that right now the county and state maps just plot the conterminous lower 48. The map of individual occurrences shows the lower 48 + Alaska</h3>

<div class="highlight"><pre><code class="r"><span class="c1"># By county</span>
bisonmap<span class="p">(</span>out<span class="p">,</span> tomap <span class="o">=</span> <span class="s">&quot;county&quot;</span><span class="p">)</span>
</code></pre></div>


<p><img src="/public/img/2013-05-25-rbison/map11.png" alt="center" /></p>

<div class="highlight"><pre><code class="r"><span class="c1"># By state</span>
bisonmap<span class="p">(</span>out<span class="p">,</span> tomap <span class="o">=</span> <span class="s">&quot;state&quot;</span><span class="p">)</span>
</code></pre></div>


<p><img src="/public/img/2013-05-25-rbison/map12.png" alt="center" /></p>

<div class="highlight"><pre><code class="r"><span class="c1"># Individual locations</span>
bisonmap<span class="p">(</span>out<span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">## Rendering map...plotting 199 points
</code></pre></div>


<p><img src="/public/img/2013-05-25-rbison/map13.png" alt="center" /></p>

<hr />

<h3>When plotting occurrences, you can pass additional arguments into the <code>bisonmap</code> function.</h3>

<h4>For example, you can jitter the points</h4>

<div class="highlight"><pre><code class="r">bisonmap<span class="p">(</span>input <span class="o">=</span> out<span class="p">,</span> geom <span class="o">=</span> geom_jitter<span class="p">)</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">## Rendering map...plotting 199 points
</code></pre></div>


<p><img src="/public/img/2013-05-25-rbison/map2.png" alt="center" /></p>

<h4>And you can specify by how much you want the points to jitter (here an extreme example to make it obvious)</h4>

<div class="highlight"><pre><code class="r">library<span class="p">(</span>ggplot2<span class="p">)</span>
bisonmap<span class="p">(</span>input <span class="o">=</span> out<span class="p">,</span> geom <span class="o">=</span> geom_jitter<span class="p">,</span> jitter <span class="o">=</span> position_jitter<span class="p">(</span>width <span class="o">=</span> <span class="m">5</span><span class="p">))</span>
</code></pre></div>




<div class="highlight"><pre><code class="text">## Rendering map...plotting 199 points
</code></pre></div>


<p><img src="/public/img/2013-05-25-rbison/map3.png" alt="center" /></p>

<hr />

<h4>Let us know if you have any feature requests or find bugs at <a href="https://github.com/ropensci/rbison/issues">our GitHub Issues page</a>.</h4>

  </div>
  
</div>

<!-- Pagination links -->
<div class="pagination">
  
    <a href="/page8" class="older">Older</a>
  
  
    
      <a href="/page6" class="newer">Newer</a>
    
  
</div>
    </div>

  </body>

  <footer>  
  <!-- Disqus code -->
  <script type="text/javascript">
      /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
      var disqus_shortname = 'recology'; // required: replace example with your forum shortname

      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function () {
          var s = document.createElement('script'); s.async = true;
          s.type = 'text/javascript';
          s.src = 'http://' + disqus_shortname + '.disqus.com/count.js';
          (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
      }());
  </script>

  <!-- Gaug.es Analytics -->
  <script type="text/javascript">
    var _gauges = _gauges || [];
    (function() {
      var t   = document.createElement('script');
      t.type  = 'text/javascript';
      t.async = true;
      t.id    = 'gauges-tracker';
      t.setAttribute('data-site-id', '4efd83a6f5a1f5158a000004');
      t.src = '//secure.gaug.es/track.js';
      var s = document.getElementsByTagName('script')[0];
      s.parentNode.insertBefore(t, s);
    })();
  </script>
</footer>      
</html>