<!DOCTYPE html>

<html>
  <head>

    <meta http-equiv="Content-type" content="text/html;charset=UTF-8">

    <title>Recology: An exploration of using R for ecology, evolution, and open science.</title>

    <script type="text/javascript" src="http://use.typekit.com/jjw2jrj.js"></script>
    <script type="text/javascript">try{Typekit.load();}catch(e){}</script>

    <link type="text/css" rel="stylesheet" href="/stylesheets/reset.css">
    <link type="text/css" rel="stylesheet" href="/stylesheets/screen.css">
    <link type="text/css" rel="stylesheet" href="/stylesheets/code.css">
    <link rel="alternate" type="application/atom+xml"
      title="User Obsessed Feed" href="http://feeds.feedburner.com/userobsessed" />

    <link rel="apple-touch-icon" href="/images/iOS-114.png" />
    <link rel="apple-touch-icon-precomposed" href="/images/iOS-57.png" />
    <link rel="apple-touch-icon-precomposed" sizes="72x72" media="screen and (resolution: 132dpi)" href="/images/iOS-72.png" />
    <link rel="apple-touch-icon-precomposed" sizes="114x114" media="screen and (resolution: 326dpi)" href="/images/iOS-114.png" />

    <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0" />



  </head>
  <body class="default">

    <section id="main">

      <section id="masthead" class="group">
        <header class="group">
          <p><a href="/">Recology</a></p>
          <h1>An exploration of using R for ecology, evolution, and open science.</h1>
          <img src="/images/uo-logo.png">
        </header>
      </section>

      <section id="focus" class="group">

        <section id="content" class="group">
          <section id="posts">
  <ol>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/09/06/jonathan-eisen-on-the-fourth-domain-and-open-science/">Jonathan Eisen on the Fourth Domain and Open Science</a></h2>
  </header>
  
    <iframe frameborder="0" height="227" src="http://player.vimeo.com/video/28444926?portrait=0" width="400"></iframe><br /><br /><br /><a href="http://vimeo.com/28444926">Stalking the Fourth Domain with Jonathan Eisen, Ph D</a> from <a href="http://vimeo.com/user8340111">mendelspod</a> on <a href="http://vimeo.com/">Vimeo</a>.

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-09-06T12:00:00-05:00">Tuesday, September  6</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F09%2F06%2Fjonathan-eisen-on-the-fourth-domain-and-open-science%2F&amp;title=Jonathan%20Eisen%20on%20the%20Fourth%20Domain%20and%20Open%20Science" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/08/31/rnpn-an-r-interface-for-the-national-phenology-network/">rnpn: An R interface for the National Phenology Network</a></h2>
  </header>
  
    The team at <a href="http://ropensci.org/">rOpenSci</a> and I have been working on a wrapper for the <a href="http://www.usanpn.org/">USA National Phenology Network</a> API. The following is a demo of some of the current possibilities. We will have more functions down the road. Get the publicly available code, and contribute, at Github <a href="https://github.com/ropensci/rnpn">here</a>. If you try this out look at the <a href="https://github.com/ropensci/rnpn/blob/master/DESCRIPTION">Description file </a>for the required R packages to run rnpn. Let us know at Github (<a href="https://github.com/ropensci">here</a>) or at our website &nbsp;<a href="http://ropensci.org/">http://ropensci.org/</a>, or&nbsp;in the comments below, or on twitter (@rOpenSci), what use cases you would like to see with the rnpn package.<br /><br /><b><u>Method and demo of each</u></b>:<br /><i><b>Get observations for species by day</b></i><br /><i><span class="Apple-style-span" style="font-style: normal;">From the documentation: "</span></i><span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function will return a list of species, containing all the dates which observations were made about the species, and a count of the number of such observations made on that date.</span><i><span class="Apple-style-span" style="font-style: normal;">"</span></i><br /><br /><span class="Apple-style-span" style="font-size: x-small;">#### Note, the data below is truncated for blogging brevity...</span><br /><div style="overflow: auto;"><div class="geshifilter"><pre class="r geshifilter-R" style="font-family: monospace;">&gt; getobsspbyday<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">2</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: blue;">'2008-01-01'</span><span style="color: #339933;">,</span> <span style="color: blue;">'2011-12-31'</span><span style="color: #666666;">) # Searched for species 1 and 2 from Jan 1, 2008 to Dec 31, 2011</span><br />          <a href="http://inside-r.org/r-doc/base/date"><span style="color: #003399; font-weight: bold;">date</span></a> count   species<br /><span style="color: #cc66cc;">1</span>   <span style="color: #cc66cc;">2009</span>-03-08     <span style="color: #cc66cc;">2</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">2</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">15</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">3</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">22</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">4</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">24</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">5</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">26</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">6</span>   <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">17</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">7</span>   <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">24</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">8</span>   <span style="color: #cc66cc;">2009</span>-05-<span style="color: #cc66cc;">12</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">9</span>   <span style="color: #cc66cc;">2009</span>-05-<span style="color: #cc66cc;">20</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">10</span>  <span style="color: #cc66cc;">2009</span>-<span style="color: #cc66cc;">11</span>-<span style="color: #cc66cc;">24</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">11</span>  <span style="color: #cc66cc;">2009</span>-<span style="color: #cc66cc;">12</span>-07     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">12</span>  <span style="color: #cc66cc;">2010</span>-01-<span style="color: #cc66cc;">18</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">13</span>  <span style="color: #cc66cc;">2010</span>-01-<span style="color: #cc66cc;">23</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">62</span>  <span style="color: #cc66cc;">2011</span>-05-<span style="color: #cc66cc;">29</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">63</span>  <span style="color: #cc66cc;">2011</span>-06-<span style="color: #cc66cc;">27</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">64</span>  <span style="color: #cc66cc;">2011</span>-06-<span style="color: #cc66cc;">30</span>     <span style="color: #cc66cc;">2</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">65</span>  <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">17</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">66</span>  <span style="color: #cc66cc;">2009</span>-04-03     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">67</span>  <span style="color: #cc66cc;">2009</span>-04-05     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">68</span>  <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">10</span>     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">69</span>  <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">17</span>     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span></pre></div></div><br /><br /><br /><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><b>Get individuals at specific stations</b></i><br />From the documentation: "<span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function returns all of the individuals at a series of stations.</span>"<br /><div><br /><div style="overflow-x: auto; overflow-y: auto;"><div class="geshifilter"><pre class="r geshifilter-R" style="font-family: monospace;">&gt; getindsatstations<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">507</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">523</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span> <span style="color: #666666;"># Searched for any individuals at stations 507 and 523</span><br />   individual_id individual_name species_id kingdom<br /><span style="color: #cc66cc;">1</span>           <span style="color: #cc66cc;">1200</span>         dogwood         <span style="color: #cc66cc;">12</span> Plantae<br /><span style="color: #cc66cc;">2</span>           <span style="color: #cc66cc;">1197</span>    purple lilac         <span style="color: #cc66cc;">36</span> Plantae<br /><span style="color: #cc66cc;">3</span>           <span style="color: #cc66cc;">1193</span>         white <a href="http://inside-r.org/r-doc/base/t"><span style="color: #003399; font-weight: bold;">t</span></a>         <span style="color: #cc66cc;">38</span> Plantae<br /><span style="color: #cc66cc;">4</span>           <span style="color: #cc66cc;">3569</span>     forsythia-<span style="color: #cc66cc;">1</span>         <span style="color: #cc66cc;">73</span> Plantae<br /><span style="color: #cc66cc;">5</span>           <span style="color: #cc66cc;">1206</span>            jack        <span style="color: #cc66cc;">150</span> Plantae<br /><span style="color: #cc66cc;">6</span>           <span style="color: #cc66cc;">1199</span>      trout lily        <span style="color: #cc66cc;">161</span> Plantae<br /><span style="color: #cc66cc;">7</span>           <span style="color: #cc66cc;">1198</span>           dandy        <span style="color: #cc66cc;">189</span> Plantae<br /><span style="color: #cc66cc;">8</span>           <span style="color: #cc66cc;">1192</span>           red <a href="http://inside-r.org/r-doc/base/t"><span style="color: #003399; font-weight: bold;">t</span></a>        <span style="color: #cc66cc;">192</span> Plantae<br /><span style="color: #cc66cc;">9</span>           <span style="color: #cc66cc;">1710</span>    common lilac         <span style="color: #cc66cc;">36</span> Plantae<br /><span style="color: #cc66cc;">10</span>          <span style="color: #cc66cc;">1711</span>  common lilac <span style="color: #cc66cc;">2</span>         <span style="color: #cc66cc;">36</span> Plantae<br /><span style="color: #cc66cc;">11</span>          <span style="color: #cc66cc;">1712</span>       dandelion        <span style="color: #cc66cc;">189</span> Plantae</pre></div></div><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><b>Get individuals of species at stations</b></i><br />From the documentation:&nbsp;"<span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function will return a list of all the individuals, which are members of a species, among &nbsp;any number of stations.</span>"</div><div><br /><div style="overflow-x: auto; overflow-y: auto;"><div class="geshifilter"><pre class="r geshifilter-R" style="font-family: monospace;">&gt; getindspatstations<span style="color: #009900;">(</span><span style="color: #cc66cc;">35</span><span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">60</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">259</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">2009</span><span style="color: #009900;">)</span>  <span style="color: #666666;"># Search for individuals of species 35 at stations 60 and 259 in year 2009</span><br />  individual_id individual_name number_observations<br /><span style="color: #cc66cc;">1</span>          <span style="color: #cc66cc;">1715</span>            west                   <span style="color: #cc66cc;">5</span><br /><span style="color: #cc66cc;">2</span>          <span style="color: #cc66cc;">1716</span>            east                   <span style="color: #cc66cc;">5</span></pre></div></div><br /><br /><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><b>Get observation associated with particular observation</b></i><br /><span class="Apple-style-span"><span class="Apple-style-span" style="font-style: normal;">From the documentation:</span><span class="Apple-style-span" style="font-family: Arial; font-size: x-small;"><span class="Apple-style-span" style="white-space: pre-wrap;"><span class="Apple-style-span" style="font-style: normal;">&nbsp;</span>"</span></span></span><span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function will return the comment associated with a particular observation.</span><span class="Apple-style-span"><span class="Apple-style-span" style="font-family: Arial; font-size: x-small;"><span class="Apple-style-span" style="white-space: pre-wrap;">"</span></span></span><br /><span class="Apple-style-span" style="font-family: Arial; font-size: x-small;"><span class="Apple-style-span" style="white-space: pre-wrap;"></span></span><br /><span class="Apple-style-span" style="font-family: monospace; white-space: pre;">&gt; getobscomm<span style="color: #009900;">(</span><span style="color: #cc66cc;">1938</span><span style="color: #009900;">)</span> <span style="color: #666666;"># The observation for observation number 1938</span></span><br /><div style="overflow-x: auto; overflow-y: auto;"><div class="geshifilter"><pre class="r geshifilter-R" style="font-family: monospace;">$observation_comment<br /><span style="color: #009900;">[</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span> <span style="color: blue;">"some lower branches are bare"</span></pre></div></div></div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-08-31T12:00:00-05:00">Wednesday, August 31</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F08%2F31%2Frnpn-an-r-interface-for-the-national-phenology-network%2F&amp;title=rnpn%3A%20An%20R%20interface%20for%20the%20National%20Phenology%20Network" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/08/22/tenure-track-position-in-systematics-at-the-university-of-vermont/">Tenure track position in systematics at the University of Vermont</a></h2>
  </header>
  
    <span class="Apple-style-span" style="background-color: white; color: #4f4f4f; font-family: Arial, Tahoma, Helvetica, FreeSans, sans-serif; font-size: 15px; line-height: 20px;"></span><br /><center style="text-align: left;">There is an awesome position opening up for an assistant professor in systematics at the University of Vermont. Below is the announcement, and see the <a href="http://currentecology.blogspot.com/2011/08/tenure-track-position-in-systematics-at.html">original post</a> at the <a href="http://currentecology.blogspot.com/">Distributed Ecology blog</a>. Why is this related to R? One can do a lot of systematics work in R, including retrieving scientific collections data through an upcoming package handshaking with <a href="http://vertnet.org/index.php">VertNet</a> (part of the <a href="http://ropensci.org/">rOpenSci</a> project), managing large data sets, retrieval of GenBank data through the ape package (see fxn read.genbank), phylogenetic reconstruction and analysis, and more. So I am sure a systematist with R ninja skills will surely have a head up on the rest of the field.&nbsp;</center><center style="text-align: left;"><br /></center><center style="text-align: left;"><br /></center><center><b><br /></b></center><center><b>Assistant Professor in Systematics</b></center><br /><br />Department of Biology<br />University of Vermont<br />Burlington, Vermont<br /><br />The Department of Biology of the University of Vermont seeks applications for a tenure- track Assistant Professor position in Systematics and Evolutionary Biology of arthropods, especially insects. The position will be open in the fall of 2012. The successful candidate will have expertise in classical and molecular systematics, including analysis of complex data sets. Candidates pursuing phylogenomics and innovative methods in bioinformatics in combination with taxonomy are especially encouraged to apply. Department information at:<a href="http://www.uvm.edu/~biology/" style="color: #0499fa; text-decoration: none;">&nbsp;http://www.uvm.edu/~biology/</a>.<br /><br /><br />All applicants are expected to: 1) hold a Ph.D. degree in relevant disciplines and have two or more years of postdoctoral experience; 2) develop a competitively funded research program; 3) teach undergraduate courses (chosen from among general biology, evolution, systematic entomology, and others in the candidate's expertise); 4) teach, mentor and advise undergraduate and graduate students; and 5) oversee a natural history collection of historic significance.<br /><br />Candidates must apply online:<a href="http://www.uvmjobs.com/" style="color: #0499fa; text-decoration: none;">&nbsp;http://www.uvmjobs.com/</a>. On left see "Search Postings" then find "Biology" under "HCM Department" then posting 0040090 (first posting). Sorry, but we cannot supply the direct link because it will time out.<br /><br />Attach a cover letter with a statement of research focus and teaching interests (one document), a curriculum vitae, representative publications, and the contact information of three references.<br /><br />Review of applications will begin on September 15, 2011, and will continue until the position is filled. Questions and up to three additional publications may be directed to Dr. Jos. J. Schall: jschall@uvm.edu.<br /><br />The University of Vermont recently identified several "Spires of Excellence" in which it will strategically focus institutional investments and growth over the next several years. One spire associated with the position is Complex Systems. Candidates whose research interests align with this spire are especially encouraged to apply<a href="http://www.uvm.edu/~tri/" style="color: #0499fa; text-decoration: none;">http://www.uvm.edu/~tri/</a>.<br />The University seeks faculty who can contribute to the diversity and excellence of the academic community through their research, teaching, and/or service. Applicants are requested to include in their cover letter information about how they will further this goal. The University of Vermont is an Affirmative Action/Equal Opportunity employer. The Department is committed to increasing faculty diversity and welcomes applications from women, underrepresented ethnic, racial and cultural groups, and from people with disabilities.

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-08-22T12:00:00-05:00">Monday, August 22</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F08%2F22%2Ftenure-track-position-in-systematics-at-the-university-of-vermont%2F&amp;title=Tenure%20track%20position%20in%20systematics%20at%20the%20University%20of%20Vermont" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/08/12/thursday-at-esa11/">Thursday at #ESA11</a></h2>
  </header>
  
    Interesting talks/posters:<div><br />Richard Lankau presented research on trade-offs and competitive ability. He suggests that during range expansion selection for increased intraspecific competitive ability in older populations leads to loss of traits for interspecific competitive traits due to trade-offs between these traits. <br /><br /><br />Ellner emphatically states that rapid evolution DOES matter for ecological responses, and longer-term evolutionary patterns as well. [His paper on the talk he was giving came out prior to his talk, which he pointed out, good form sir]<br /><br /><br />Lauren Sullivan gave an interesting talk on bottom up and top down effects on plant reproduction in one site of a huge network of sites doing similar nutrient and herbivory manipulations around the globe - NutNet (go here: <a href="http://nutnet.science.oregonstate.edu/">http://nutnet.science.oregonstate.edu/</a>). <br /><br /><br />Laura Prugh shows in California that the engineering effects (i.e., the mounds that they make) of giant kangaroo rats are more important for the associated food web than the species interaction effects (the proxy used was just density of rats). <br /><br /><br />Kristy Deiner suggests that chironomids are more phylogenetic similar in lakes with stocked fish relative to fishless lakes, in high elevation lakes in the Sierra Nevada. She used barcode data to generate her phylogeny of chironomids. If you have barcode data and want to search BOLD Systems site, one option is doing it from R using rbold, a package under development at <a href="http://ropensci.org/">rOpenSci</a> (code at <a href="https://github.com/SChamberlain/rbold">Github</a>). <br /><br /><br />Jessica Gurevitch presented a large working group's methods/approach to a set of reviews on invasion biology. We didn't get to see a lot of results from this work, but I personally was glad to see her explaining to a packed room the utility of meta-analysis, and comparing to the medical field in which meta-analysis is sort of the gold standard by which to draw conclusions. <br /><br /><br />Following Jessica, Jason Fridley told us about the Evolutionary Imbalance Hypothesis (EIH) (see my notes <a href="http://schamber.wordpress.com/lab-notebook-2/">here</a>). He posed the problem of, when two biotas come together, what determines which species are retained in this new community and which species are left out. He listed a litany of traits/responses to measure to get at this problem, but suggested that with a little bit of "desktop ecology", we could simply ask: Is the invasability of X region related to the phylogenetic diversity of that region? In three destination regions (Eastern Deciduous Forests, Mediterranean California, and the Czech Republic) out of four there was a positive relationship between proportion of invasive plant species in a source region and the phylogenetic diversity of the source regions. <ul></ul></div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-08-12T12:00:00-05:00">Friday, August 12</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F08%2F12%2Fthursday-at-esa11%2F&amp;title=Thursday%20at%20%23ESA11" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/08/11/wednesday-at-esa11/">Wednesday at #ESA11</a></h2>
  </header>
  
    <span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Interesting talks/posters:</span><br /><div><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;"><br /></span></div><div><ul><li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Ethan White's poster describing <a href="http://ecologicaldata.org/">EcologicalData.org</a> was of course awesome given my interest in getting data into the hands of ecologists over at <a href="http://ropensci.org/">rOpenSci.org</a>. Ethan also has software you can download on your machine to get the datasets you want easily - <a href="http://ecologicaldata.org/ecodata-retriever">EcoData Retriever</a>. [rOpenSci will try to take advantage of their work and allow you to call the retriever from R]</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Carl Boettiger's talk was awesome. He explained how we need better tools to be able to predict collapses using early warning signals. He developed a way to estimate the statistical distribution of probabilities of system collapse.&nbsp;</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Jennifer Dunne: Explained how she put together an ancient network from Germany. Bravo.&nbsp;</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Carlos Melian explained his model of network buildup that starts from individuals, allows speciation, and other evolutionary processes.&nbsp;</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;"><span class="Apple-style-span" style="background-color: white;">Rachel Winfree told us that in two sets of mutualistic plant-pollinator networks in New Jersey and California, that the least connected pollinator species were the most likely to be lost from the network with increasing agricultural intensity.&nbsp;</span></span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;"><span class="Apple-style-span" style="background-color: white;">Dan Cariveau suggests that pollination crop services can be stabilized even with increasing agriculture intensity if in fact pollinator species respond in different ways. That is, some pollinators may decrease in abundance with increasing ag intensity, while other species may increase - retaining overall pollination services to crops.</span></span></li>
</ul></div><div><ul></ul></div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-08-11T12:00:00-05:00">Thursday, August 11</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F08%2F11%2Fwednesday-at-esa11%2F&amp;title=Wednesday%20at%20%23ESA11" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/08/08/monday-at-esa11/">Monday at #ESA11</a></h2>
  </header>
  
    Monday was a good day at ESA in Austin.&nbsp;There were a few topics I promised to report on in my blogging/tweeting.<br /><br /><br />...focused on open source data. <a href="http://www.nceas.ucsb.edu/~strasser/Site/Home.html">Carly Strasser's</a> presentation on guidelines for data management was awesome (including other talks in the symposium on Creating Effective Data Management Plans for Ecological Research). Although this was a good session, I can't&nbsp;help but wish that they had hammered home the need for open science more. Oh well. Also, they talked&nbsp;a lot about how, and not a lot of why we should properly curate data. Still, a good session. One issue Carly and I talked about was tracking code in versioning systems such as <a href="https://github.com/">Github</a>. There doesn't seem to be a culture of versioning code for analyses/simulations in ecology. But when we get there...it will be easier to share/track/collaborate on &nbsp;code.<br /><br />...used R software. <a href="http://eco.confex.com/eco/2011/preliminaryprogram/abstract_31982.htm">David Jennings</a>&nbsp;talked about a meta-analysis asking if phylogenetic distance influences competition strength in pairwise experiments. David used the metafor package in R to do his meta-analysis. Good form sir.<br /><br />...did cool science. <a href="http://www.ecologicalevolution.org/people/former-lab-members/matthew-helmus/">Matt Helmus </a>presented a great talk on phylogenetic species area curves (likely using R, or Matlab maybe?).<br /><br /><br />p.s. <a href="http://ropensci.org/developers/">We</a> launched <a href="http://ropensci.org/2011/07/welcome-to-ropensci/">rOpenSci</a> today.<br /><br /><br />Oddities:<br /><ul><li>The Tilman effect - Tilman's talk was so packed it looked like there was a line waiting to get into a trendy bar. <a href="http://www.math.ualberta.ca/~ashander/images/til.jpg">Here's a picture</a>&nbsp;(credit: Jaime Ashander). Bigger room next time anyone?&nbsp;</li>
<li>Wiley came out with an open source journal called <a href="http://onlinelibrary.wiley.com/journal/10.1002/(ISSN)2045-7758">Ecology and Evolution</a>. This brings them to 3 open source journals (the other two are in other fields). We (<a href="http://ropensci.org/">rOpenSci</a>) will attempt to hand-shake with these journals.&nbsp; </li>
<li>The vegetarian lunch option was surprisingly good. Nice.&nbsp;</li>
</ul>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-08-08T12:00:00-05:00">Monday, August  8</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F08%2F08%2Fmonday-at-esa11%2F&amp;title=Monday%20at%20%23ESA11" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/08/08/-esa11-ropensci-a-collaborative-effort-to-develop-r-based-tools-for-facilitating-open-science/">(#ESA11) rOpenSci: a collaborative effort to develop R-based tools for facilitating Open Science</a></h2>
  </header>
  
    Our&nbsp;<a href="http://ropensci.org/developers/">development team</a>&nbsp;would like to announce the launch of&nbsp;<a href="http://ropensci.org/">rOpenSci</a>. As the title states, this project aims to create&nbsp;<a href="http://www.r-project.org/">R</a>&nbsp;<a href="http://cran.r-project.org/web/packages/available_packages_by_name.html">packages</a>&nbsp;to make open science more available to researchers.<br /><br /><table align="center" cellpadding="0" cellspacing="0" class="tr-caption-container" style="margin-left: auto; margin-right: auto; text-align: center;"><tbody>
<tr><td style="text-align: center;"><a href="http://1.bp.blogspot.com/-9YIk1e1liUU/Tj8YAoOPyQI/AAAAAAAAElQ/9gCMQ8CJORI/s1600/ropensci.png" imageanchor="1" style="margin-left: auto; margin-right: auto;"><img border="0" src="http://1.bp.blogspot.com/-9YIk1e1liUU/Tj8YAoOPyQI/AAAAAAAAElQ/9gCMQ8CJORI/s1600/ropensci.png" /></a></td></tr>
<tr><td class="tr-caption" style="font-size: 13px; text-align: center;"><a href="http://ropensci.org/">http://ropensci.org/</a></td></tr>
</tbody></table><br />What this means is that we seek to connect researchers using R with as much open data as possible, mainly through APIs. There are a number of R packages that already do this (e.g.,&nbsp;<a href="http://cran.r-project.org/web/packages/infochimps/index.html">infochimps</a>,&nbsp;<a href="http://cran.r-project.org/web/packages/twitteR/index.html">twitteR</a>), but we are making more packages, e.g., for&nbsp;<a href="https://github.com/cboettig/RMendeley">Mendeley</a>,&nbsp;<a href="https://github.com/SChamberlain/rplos">PLoS Journals</a>, and&nbsp;<a href="https://github.com/SChamberlain/taxize_">taxonomic sources</a>&nbsp;(<a href="http://www.itis.gov/">ITIS</a>,&nbsp;<a href="http://www.eol.org/">EOL</a>,&nbsp;<a href="http://tnrs.iplantcollaborative.org/">TNRS</a>,&nbsp;<a href="http://www.phylodiversity.net/phylomatic/">Phylomatic</a>,&nbsp;<a href="http://www.ubio.org/">UBio</a>).<br /><br />Importantly, we are creating a package called&nbsp;<a href="http://ropensci.org/">rOpenSci</a>, which aims to integrate functions from packages for individual open data sources.<br /><br />If you are somewhat interested, follow our progress on our&nbsp;<a href="http://ropensci.org/">website</a>, on&nbsp;<a href="http://twitter.com/#!/rOpenSci">Twitter</a>, or&nbsp;<a href="http://ropensci.org/contact/">contact us</a>. If you are really^2 interested you could go to&nbsp;<a href="https://github.com/">Github</a>&nbsp;and contribute. If &nbsp;you are really^3 interested,&nbsp;<a href="http://ropensci.org/developers/">join our development team</a>.

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-08-08T12:00:00-05:00">Monday, August  8</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F08%2F08%2F-esa11-ropensci-a-collaborative-effort-to-develop-r-based-tools-for-facilitating-open-science%2F&amp;title=(%23ESA11)%20rOpenSci%3A%20a%20collaborative%20effort%20to%20develop%20R-based%20tools%20for%20facilitating%20Open%20Science" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/07/31/blogging-tweeting-from-esa11/">Blogging/tweeting from #ESA11</a></h2>
  </header>
  
    I will be blogging about the upcoming Ecological Society of America meeting in Austin, TX. I will focus on discussing talks/posters that:<br /><br /><ol><li>Have taken a cool approach to using data, or</li><li>Have focused on open science/data, or</li><li>Done&nbsp;something cool with R software, or</li><li>Are just exciting in general</li></ol><div><br /></div><div>I will also tweet throughout the meeting from @recology_ (yes the underscore is part of the name, recology was already taken).&nbsp;</div><div><br /></div><div>The hashtag for the meeting this year is #ESA11</div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-07-31T12:00:00-05:00">Sunday, July 31</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F07%2F31%2Fblogging-tweeting-from-esa11%2F&amp;title=Blogging%2Ftweeting%20from%20%23ESA11" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/07/18/models-in-evolutionary-ecology-seminar-organized-by-timothee-poisot/">Models in Evolutionary Ecology seminar, organized by Timothee Poisot</a></h2>
  </header>
  
    Here is one of the talks by Thomas Brouquet, and see the rest <a href="http://www.dailymotion.com/playlist/x1mv7l_mez_en_video_cefe-colloque#videoId=xjylea">here</a>. <br /><br /><br /><iframe frameborder="0" height="270" src="http://www.dailymotion.com/embed/video/xjylea" width="480"></iframe><br /><a href="http://www.dailymotion.com/video/xjylea_thomas-broquet_tech" target="_blank">Thomas Broquet</a> <i>by <a href="http://www.dailymotion.com/mez_en_video" target="_blank">mez_en_video</a></i>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-07-18T12:00:00-05:00">Monday, July 18</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F07%2F18%2Fmodels-in-evolutionary-ecology-seminar-organized-by-timothee-poisot%2F&amp;title=Models%20in%20Evolutionary%20Ecology%20seminar%2C%20organized%20by%20Timothee%20Poisot" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/07/15/archiving-ecology-evolution-data-sets-online/">Archiving ecology/evolution data sets online</a></h2>
  </header>
  
    <span class="Apple-style-span" style="font-family: Georgia, 'Times New Roman', 'Bitstream Charter', Times, serif; font-size: 13px; line-height: 19px;"></span><br />We now have many options for archiving data sets online:<br /><br /><a data-mce-href="http://datadryad.org/" href="http://datadryad.org/" target="_blank">Dryad</a>,&nbsp;<a data-mce-href="http://knb.ecoinformatics.org/index.jsp" href="http://knb.ecoinformatics.org/index.jsp" target="_blank">KNB</a>,&nbsp;<a data-mce-href="http://www.esapubs.org/archive/" href="http://www.esapubs.org/archive/" target="_blank">Ecological Archives</a>,&nbsp;<a data-mce-href="http://www.esapubs.org/archive/archive_D.htm" href="http://www.esapubs.org/archive/archive_D.htm" target="_blank">Ecology Data Papers</a>,&nbsp;<a data-mce-href="http://ecologicaldata.org/" href="http://ecologicaldata.org/" target="_blank">Ecological Data</a>, etc.<br /><br />However, these portals largely do not communicate with one another as far as I know, and there is no way to search over all data set sources, again, as far as I know. So, I wonder if it would ease finding of all these&nbsp;different data sets to get these different sites to get their data sets cloned on a site like Infochimps, or have links from Infochimps. &nbsp;Infochimps already has APIs (and there's an R wrapper for the Infochimps API already set up here: http://cran.r-project.org/web/packages/infochimps/index.html by Drew Conway), and they have discussions set up there, etc.<br /><br />Does it make sense to post data sets linked to published works on Infochimps? I think probably not know that I think about it. But perhaps it makes sense for other data sets, or subsets of data sets that are not linked with published works to be posted there as I know at least Dryad only accepts data sets linked with published papers.<br /><br />One use case is there was a tweet from someone recently that his students were excited about getting their data sets on their resume/CV, but didn't think there was a way to put them any place where there wasn't a precondition that the data set was linked with a published work. Seems like this&nbsp;could be a good opportunity to place these datasets on Infcohimps, and at least they are available then where a lot of people are searching for data sets, etc.<br /><br />What I think would be ideal is if Dryad, KNB, etc. could link their datasets to Infochimps, where they could be found, then users can either get them from Infochimps, or perhaps you would have to go to the Dryad site, e.g. But at least you could search over all ecological data sets then.

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-07-15T12:00:00-05:00">Friday, July 15</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F07%2F15%2Farchiving-ecology-evolution-data-sets-online%2F&amp;title=Archiving%20ecology%2Fevolution%20data%20sets%20online" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
  </ol>

  
  <nav id="pagination">
    <ol>
      
        
      
      
      <li>
        <a href="/pagelayoutdefaulturl/page4/index.htmlcontent<section id="posts">
  <ol>
    {% for post in paginator.posts %}
    <li>
      {% include post.html %}
    </li>
    {% endfor %}
  </ol>

  {% if paginator.total_pages > 1 %}
  <nav id="pagination">
    <ol>
      {% if paginator.previous_page > 1 %}
        {% capture page_url %}page{{ page }}{% endcapture%}
      {% endif %}
      {% if paginator.previous_page %}
      <li>
        <a href="/{{ page_url }}" title="Go to previous page">
          &laquo;
        </a>
      </li>
      {% endif %}
      <li>
      {% for page in (1..paginator.total_pages) %}
      <li>
        {% if page > 1 %}
          {% capture page_url %}page{{ page }}{% endcapture%}
        {% endif %}
        {% if page == paginator.page %}
          {{ page }}
        {% else %}
          <a href="/{{ page_url }}">{{ page }}</a>
        {% endif %}
      </li>
      {% endfor %}
      {% if paginator.next_page %}
      <li>
        <a href="/page{{ paginator.next_page }}" title="Go to next page">
          &raquo;
        </a>
      </li>
      {% endif %}
    </ol>
  </nav>
  {% endif %}
</section>
" title="Go to previous page">
          &laquo;
        </a>
      </li>
      
      <li>
      
      <li>
        
        
          <a href="/pagelayoutdefaulturl/page4/index.htmlcontent<section id="posts">
  <ol>
    {% for post in paginator.posts %}
    <li>
      {% include post.html %}
    </li>
    {% endfor %}
  </ol>

  {% if paginator.total_pages > 1 %}
  <nav id="pagination">
    <ol>
      {% if paginator.previous_page > 1 %}
        {% capture page_url %}page{{ page }}{% endcapture%}
      {% endif %}
      {% if paginator.previous_page %}
      <li>
        <a href="/{{ page_url }}" title="Go to previous page">
          &laquo;
        </a>
      </li>
      {% endif %}
      <li>
      {% for page in (1..paginator.total_pages) %}
      <li>
        {% if page > 1 %}
          {% capture page_url %}page{{ page }}{% endcapture%}
        {% endif %}
        {% if page == paginator.page %}
          {{ page }}
        {% else %}
          <a href="/{{ page_url }}">{{ page }}</a>
        {% endif %}
      </li>
      {% endfor %}
      {% if paginator.next_page %}
      <li>
        <a href="/page{{ paginator.next_page }}" title="Go to next page">
          &raquo;
        </a>
      </li>
      {% endif %}
    </ol>
  </nav>
  {% endif %}
</section>
">1</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page2">2</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page3">3</a>
        
      </li>
      
      <li>
        
          
        
        
          4
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page5">5</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page6">6</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page7">7</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page8">8</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page9">9</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page10">10</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page11">11</a>
        
      </li>
      
      
      <li>
        <a href="/page5" title="Go to next page">
          &raquo;
        </a>
      </li>
      
    </ol>
  </nav>
  
</section>

        </section>

        <section id="sidebar">

          <section id="meta-col1" class="meta group">
            <aside id="about">
              <header>
                <h3>About</h3>
              </header>
              <p>
                Recology is a blog that explores using R for ecology, evolution, and open science. <a href="/about">See more.</a>
              </p>
            </aside>

            <aside>
              <header>
                <h3>Stay up to date</h3>
              </header>
              <nav>
                <ul>
                  <li class="icon rss"><a href="http://feeds.feedburner.com/XX">RSS</a></li>
                  <li class="icon twitter"><a href="http://twitter.com/recology_">Twitter</a></li>
                  <li class="icon"><a href="/subscribe">More</a></li>
                </ul>
              </nav>
            </aside>

          </section>

          <section id="meta-col2" class="meta group">

            <aside id="bio">
              <header>
                <h3>The Author</h3>
              </header>
              <p>
              My name is Scott Chamberlain. I am a researcher in ecology and evolution.
              </p>
            </aside>

            <aside>
              <nav id="extra">
                <ul>
                  <li><a href="/colophon">Colophon</a></li>
                  <li><a href="/contact">Contact</a></li>
                </ul>
              </nav>
            </aside>

          </section>

        </section>

      </section>

      <section id="footer" class="group">
        <footer class="group">
          <nav id="site">
            <ul>
              <li><a href="/">Home</a></li>
              <li><a href="/about">About</a></li>
              <li><a href="/contact">Contact</a></li>
              <li><a href="/subscribe">Subscribe</a></li>
              <li><a href="/colophon">Colophon</a></li>
            </ul>
          </nav>
          <p class="copyright">
            &copy;2011 &ndash; Scott Chamberlain
          </p>
        </footer>
      </section>

    </section>

    <script type="text/javascript">

      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-23094691-1']);
      _gaq.push(['_trackPageview']);

      (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
      })();

    </script>

  </body>
</html>
