<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <title>
    
    Recology, R/etc.
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:300,400italic,400,600,700|Abril+Fatface">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/favicon.ico">
  <link rel="shortcut icon" href="/public/favicon.ico">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css" rel="stylesheet">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body class="theme-base-0f layout-reverse">

    <header class="masthead">
      <div class="masthead-inner">
        <h1>Recology</h1>
        <!-- <h1> <a href="http://recology.info/">Recology</a></h1> -->
        <p class="lead">R/etc.</p>

        <div class="colophon">
          <ul class="colophon-links">
            <li>
              <a href="/"><i class="fa fa-home fa-lg"></i></a>&nbsp;
              <a href="/about"><i class="fa fa-info-circle fa-lg"></i></a>&nbsp;
              <a href="/archives"><i class="fa fa-archive fa-lg"></i></a>&nbsp;
              <a href="/rresources"><i class="fa fa-book fa-lg"></i></a>&nbsp;
              <a href="http://rforcats.net/" rel><i class="fa fa-graduation-cap fa-lg"></i></a>&nbsp;
              <a href="feed.xml"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
              <a href="https://twitter.com/recology_"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
              <a href="/fork"><i class="fa fa-spinner fa-lg"></i></a>
            </li>
          </ul>
          <!-- <small><a href="https://github.com/mdo/hyde">Hyde</a> from <a href="https://twitter.com/mdo" target="_blank">@mdo</a>.</small> -->
        </div>
      </div>
    </header>

    <div class="content container">
      <div class="posts">
  <a style="float:right;" href="/archives" data-toggle="tooltip" data-placement="bottom" title="Archives"><i class="fa fa-archive fa-lg"></i></a>
  <a style="float:right;" href="/tags"><i class="fa fa-tags fa-lg"></i></a>&nbsp;
  
  <div class="post">
    <h1>
      <a href="/2011/08/tenure-track-position-in-systematics-at/">
        Tenure track position in systematics at the University of Vermont
      </a>
    </h1>

    <span class="post-date">22 Aug 2011</span>

    <p><span class="Apple-style-span" style="background-color: white; color: #4f4f4f; font-family: Arial, Tahoma, Helvetica, FreeSans, sans-serif; font-size: 15px; line-height: 20px;"></span><br /><center style="text-align: left;">There is an awesome position opening up for an assistant professor in systematics at the University of Vermont. Below is the announcement, and see the <a href="http://currentecology.blogspot.com/2011/08/tenure-track-position-in-systematics-at.html">original post</a> at the <a href="http://currentecology.blogspot.com/">Distributed Ecology blog</a>. Why is this related to R? One can do a lot of systematics work in R, including retrieving scientific collections data through an upcoming package handshaking with <a href="http://vertnet.org/index.php">VertNet</a> (part of the <a href="http://ropensci.org/">rOpenSci</a> project), managing large data sets, retrieval of GenBank data through the ape package (see fxn read.genbank), phylogenetic reconstruction and analysis, and more. So I am sure a systematist with R ninja skills will surely have a head up on the rest of the field.&nbsp;</center><center style="text-align: left;"><br /></center><center style="text-align: left;"><br /></center><center><b><br /></b></center><center><b>Assistant Professor in Systematics</b></center><br /><br />Department of Biology<br />University of Vermont<br />Burlington, Vermont<br /><br />The Department of Biology of the University of Vermont seeks applications for a tenure- track Assistant Professor position in Systematics and Evolutionary Biology of arthropods, especially insects. The position will be open in the fall of 2012. The successful candidate will have expertise in classical and molecular systematics, including analysis of complex data sets. Candidates pursuing phylogenomics and innovative methods in bioinformatics in combination with taxonomy are especially encouraged to apply. Department information at:<a href="http://www.uvm.edu/~biology/" style="color: #0499fa; text-decoration: none;">&nbsp;http://www.uvm.edu/~biology/</a>.<br /><br /><br />All applicants are expected to: 1) hold a Ph.D. degree in relevant disciplines and have two or more years of postdoctoral experience; 2) develop a competitively funded research program; 3) teach undergraduate courses (chosen from among general biology, evolution, systematic entomology, and others in the candidate&#39;s expertise); 4) teach, mentor and advise undergraduate and graduate students; and 5) oversee a natural history collection of historic significance.<br /><br />Candidates must apply online:<a href="http://www.uvmjobs.com/" style="color: #0499fa; text-decoration: none;">&nbsp;http://www.uvmjobs.com/</a>. On left see &quot;Search Postings&quot; then find &quot;Biology&quot; under &quot;HCM Department&quot; then posting 0040090 (first posting). Sorry, but we cannot supply the direct link because it will time out.<br /><br />Attach a cover letter with a statement of research focus and teaching interests (one document), a curriculum vitae, representative publications, and the contact information of three references.<br /><br />Review of applications will begin on September 15, 2011, and will continue until the position is filled. Questions and up to three additional publications may be directed to Dr. Jos. J. Schall: jschall@uvm.edu.<br /><br />The University of Vermont recently identified several &quot;Spires of Excellence&quot; in which it will strategically focus institutional investments and growth over the next several years. One spire associated with the position is Complex Systems. Candidates whose research interests align with this spire are especially encouraged to apply<a href="http://www.uvm.edu/~tri/" style="color: #0499fa; text-decoration: none;">http://www.uvm.edu/~tri/</a>.<br />The University seeks faculty who can contribute to the diversity and excellence of the academic community through their research, teaching, and/or service. Applicants are requested to include in their cover letter information about how they will further this goal. The University of Vermont is an Affirmative Action/Equal Opportunity employer. The Department is committed to increasing faculty diversity and welcomes applications from women, underrepresented ethnic, racial and cultural groups, and from people with disabilities.</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2011/08/thursday-at-esa11/">
        Thursday at #ESA11
      </a>
    </h1>

    <span class="post-date">12 Aug 2011</span>

    <p>Interesting talks/posters:<div><br />Richard Lankau presented research on trade-offs and competitive ability. He suggests that during range expansion selection for increased intraspecific competitive ability in older populations leads to loss of traits for interspecific competitive traits due to trade-offs between these traits. <br /><br /><br />Ellner emphatically states that rapid evolution DOES matter for ecological responses, and longer-term evolutionary patterns as well. [His paper on the talk he was giving came out prior to his talk, which he pointed out, good form sir]<br /><br /><br />Lauren Sullivan gave an interesting talk on bottom up and top down effects on plant reproduction in one site of a huge network of sites doing similar nutrient and herbivory manipulations around the globe - NutNet (go here: <a href="http://nutnet.science.oregonstate.edu/">http://nutnet.science.oregonstate.edu/</a>). <br /><br /><br />Laura Prugh shows in California that the engineering effects (i.e., the mounds that they make) of giant kangaroo rats are more important for the associated food web than the species interaction effects (the proxy used was just density of rats). <br /><br /><br />Kristy Deiner suggests that chironomids are more phylogenetic similar in lakes with stocked fish relative to fishless lakes, in high elevation lakes in the Sierra Nevada. She used barcode data to generate her phylogeny of chironomids. If you have barcode data and want to search BOLD Systems site, one option is doing it from R using rbold, a package under development at <a href="http://ropensci.org/">rOpenSci</a> (code at <a href="https://github.com/sckott/rbold">Github</a>). <br /><br /><br />Jessica Gurevitch presented a large working group&#39;s methods/approach to a set of reviews on invasion biology. We didn&#39;t get to see a lot of results from this work, but I personally was glad to see her explaining to a packed room the utility of meta-analysis, and comparing to the medical field in which meta-analysis is sort of the gold standard by which to draw conclusions. <br /><br /><br />Following Jessica, Jason Fridley told us about the Evolutionary Imbalance Hypothesis (EIH) (see my notes <a href="http://schamber.wordpress.com/lab-notebook-2/">here</a>). He posed the problem of, when two biotas come together, what determines which species are retained in this new community and which species are left out. He listed a litany of traits/responses to measure to get at this problem, but suggested that with a little bit of &quot;desktop ecology&quot;, we could simply ask: Is the invasability of X region related to the phylogenetic diversity of that region? In three destination regions (Eastern Deciduous Forests, Mediterranean California, and the Czech Republic) out of four there was a positive relationship between proportion of invasive plant species in a source region and the phylogenetic diversity of the source regions. <ul></ul></div></p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2011/08/wednesday-at-esa11/">
        Wednesday at #ESA11
      </a>
    </h1>

    <span class="post-date">11 Aug 2011</span>

    <p><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Interesting talks/posters:</span><br /><div><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;"><br /></span></div><div><ul><li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Ethan White&#39;s poster describing <a href="http://ecologicaldata.org/">EcologicalData.org</a> was of course awesome given my interest in getting data into the hands of ecologists over at <a href="http://ropensci.org/">rOpenSci.org</a>. Ethan also has software you can download on your machine to get the datasets you want easily - <a href="http://ecologicaldata.org/ecodata-retriever">EcoData Retriever</a>. [rOpenSci will try to take advantage of their work and allow you to call the retriever from R]</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Carl Boettiger&#39;s talk was awesome. He explained how we need better tools to be able to predict collapses using early warning signals. He developed a way to estimate the statistical distribution of probabilities of system collapse.&nbsp;</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Jennifer Dunne: Explained how she put together an ancient network from Germany. Bravo.&nbsp;</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;">Carlos Melian explained his model of network buildup that starts from individuals, allows speciation, and other evolutionary processes.&nbsp;</span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;"><span class="Apple-style-span" style="background-color: white;">Rachel Winfree told us that in two sets of mutualistic plant-pollinator networks in New Jersey and California, that the least connected pollinator species were the most likely to be lost from the network with increasing agricultural intensity.&nbsp;</span></span></li>
<li><span class="Apple-style-span" style="font-family: Times, 'Times New Roman', serif;"><span class="Apple-style-span" style="background-color: white;">Dan Cariveau suggests that pollination crop services can be stabilized even with increasing agriculture intensity if in fact pollinator species respond in different ways. That is, some pollinators may decrease in abundance with increasing ag intensity, while other species may increase - retaining overall pollination services to crops.</span></span></li>
</ul></div><div><ul></ul></div></p>

  </div>
  
</div>

<!-- Pagination links -->
<div class="pagination">
  
    <a href="/page38" class="older">Older</a>
  
  
    
      <a href="/page36" class="newer">Newer</a>
    
  
</div>

    </div>

    <!-- for bootstrap tooltips -->
    <script type="text/javascript">
      $("[data-toggle=\"tooltip\"]").tooltip();
    </script>

  </body>

  <footer>
  <!-- Disqus code -->
  <script type="text/javascript">
      /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
      var disqus_shortname = 'recology'; // required: replace example with your forum shortname

      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function () {
          var s = document.createElement('script'); s.async = true;
          s.type = 'text/javascript';
          s.src = 'http://' + disqus_shortname + '.disqus.com/count.js';
          (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
      }());
  </script>

  <!-- Gaug.es Analytics -->
  <script type="text/javascript">
    var _gauges = _gauges || [];
    (function() {
      var t   = document.createElement('script');
      t.type  = 'text/javascript';
      t.async = true;
      t.id    = 'gauges-tracker';
      t.setAttribute('data-site-id', '4efd83a6f5a1f5158a000004');
      t.src = '//secure.gaug.es/track.js';
      var s = document.getElementsByTagName('script')[0];
      s.parentNode.insertBefore(t, s);
    })();
  </script>
</footer>

</html>
