<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <title>
    
    Recology, R/etc.
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:300,400italic,400,600,700|Abril+Fatface">

  <link rel="stylesheet" href="/public/css/bootstrap/css/bootstrap.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/favicon.ico">
  <link rel="shortcut icon" href="/public/favicon.ico">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css" rel="stylesheet">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body class="theme-base-0f layout-reverse">

    <header class="masthead">
      <div class="masthead-inner">
        <h1>Recology</h1>
        <!-- <h1> <a href="http://recology.info/">Recology</a></h1> -->
        <p class="lead">R/etc.</p>

        <div class="colophon">
          <ul class="colophon-links">
            <li>
              <a href="/"><i class="fa fa-home fa-lg"></i></a>&nbsp;
              <a href="/about"><i class="fa fa-info-circle fa-lg"></i></a>&nbsp;
              <a href="/archives"><i class="fa fa-archive fa-lg"></i></a>&nbsp;
              <a href="/rresources"><i class="fa fa-book fa-lg"></i></a>&nbsp;
              <a href="http://rforcats.net/" rel><i class="fa fa-graduation-cap fa-lg"></i></a>&nbsp;
              <a href="/feed.xml"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
              <a href="https://twitter.com/sckottie"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
              <a href="/fork"><i class="fa fa-spinner fa-lg"></i></a>
            </li>
          </ul>
          <!-- <small><a href="https://github.com/mdo/hyde">Hyde</a> from <a href="https://twitter.com/mdo" target="_blank">@mdo</a>.</small> -->
        </div>
      </div>
    </header>

    <div class="content container">
      <div class="posts">
  <a style="float:right;" href="/archives" data-toggle="tooltip" data-placement="bottom" title="Archives"><i class="fa fa-archive fa-lg"></i></a>
  <a style="float:right;" href="/tags"><i class="fa fa-tags fa-lg"></i></a>&nbsp;
  
  <div class="post">
    <h1>
      <a href="/2012/08/ggplot-inset-map/">
        ggplot2 maps with insets
      </a>
    </h1>

    <span class="post-date">22 Aug 2012</span>

    <blockquote>
  <p>UPDATE: changed data source so that the entire example can be run by anyone on their own machine. Also, per Joachim’s suggestion, I put a box around the blown up area of the map. In addition, rgeos and maptools removed, not needed.</p>
</blockquote>

<hr />

<p>Here’s a quick demo of creating a map with an inset within it using ggplot. The inset is achieved using the <code>gridExtra</code> package.</p>

<h3 id="install-libraries">Install libraries</h3>

<pre><code class="language-r">install.packages(c("ggplot2", "maps", "grid", "gridExtra"))
</code></pre>

<pre><code class="language-r">library("ggplot2")
library("maps")
library("grid")
library("gridExtra")
</code></pre>

<h3 id="create-a-data-frame">Create a data frame</h3>

<pre><code class="language-r">dat &lt;- data.frame(ecosystem = rep(c("oak", "steppe", "prairie"), each = 8),
    lat = rnorm(24, mean = 51, sd = 1), lon = rnorm(24, mean = -113, sd = 5))
head(dat)
#&gt;   ecosystem      lat       lon
#&gt; 1       oak 49.58285 -107.6930
#&gt; 2       oak 52.58942 -116.6920
#&gt; 3       oak 50.49277 -114.5542
#&gt; 4       oak 50.05943 -116.5660
#&gt; 5       oak 51.76492 -112.1457
#&gt; 6       oak 52.82153 -112.8858
</code></pre>

<h3 id="get-maps-using-the-maps-library">Get maps using the maps library</h3>

<p>Get a map of Canada</p>

<pre><code class="language-r">canadamap &lt;- data.frame(map("world", "Canada", plot = FALSE)[c("x", "y")])
</code></pre>

<p>Get a map of smaller extent</p>

<pre><code class="language-r">canadamapsmall &lt;- canadamap[canadamap$x &lt; -90 &amp; canadamap$y &lt; 54, ]
canadamapsmall_ &lt;- na.omit(canadamapsmall)
</code></pre>

<p>This should get your corner points for the box, picking min and max of lat and lon</p>

<pre><code class="language-r">(insetrect &lt;- data.frame(xmin = min(canadamapsmall_$x), xmax = max(canadamapsmall_$x),
    ymin = min(canadamapsmall_$y), ymax = max(canadamapsmall_$y)))
#&gt;        xmin      xmax     ymin     ymax
#&gt; 1 -133.0975 -90.38942 48.04721 53.99915
</code></pre>

<h3 id="make-the-maps">Make the maps</h3>

<p>Create a theme to be used by both plots</p>

<pre><code class="language-r">ptheme &lt;- theme(
  panel.border = element_rect(colour = 'black', size = 1, linetype = 1),
  panel.grid.major = element_blank(), 
  panel.grid.minor = element_blank(),
  panel.background = element_rect(fill = 'white'),
  legend.key = element_blank()
)
</code></pre>

<p>The inset map, all of Canada</p>

<pre><code class="language-r">a &lt;- ggplot(canadamap) +
	theme_bw(base_size = 22) +
	geom_path(data = canadamap, aes(x, y), colour = "black", fill = "white") +
	geom_rect(data = insetrect, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), alpha = 0, colour = "blue", size = 1, linetype = 1) +
  ptheme %+% theme(
    legend.position = c(0.15, 0.80),
    axis.ticks = element_blank(), 
    axis.text.x = element_blank(),
    axis.text.y = element_blank()
  ) +
	labs(x = '', y = '')
</code></pre>

<p>The larger map, zoomed in, with the data</p>

<pre><code class="language-r">b &lt;- ggplot(dat, aes(lon, lat, colour = ecosystem)) +
	theme_bw(base_size = 22) +
	geom_jitter(size = 4, alpha = 0.6) +
	geom_path(data = canadamapsmall, aes(x, y), colour = "black", fill = "white") +
	scale_size(guide = "none") +
  ptheme %+% theme(
    legend.position = c(0.1, 0.20),
    legend.text = element_text(size = 12, face = 'bold'),
    legend.title = element_text(size = 12, face = 'bold'), 
    axis.ticks = element_line(size = 2)
  ) +
	labs(x = '', y = '')
</code></pre>

<h3 id="print-maps">Print maps</h3>

<p>One an inset of the other. This approach uses the <code>gridExtra</code> package for flexible alignment, etc. of ggplot graphs.</p>

<pre><code class="language-r">grid.newpage()
vpb_ &lt;- viewport(width = 1, height = 1, x = 0.5, y = 0.5)  # the larger map
vpa_ &lt;- viewport(width = 0.4, height = 0.4, x = 0.8, y = 0.8)  # the inset in upper right
print(b, vp = vpb_)
print(a, vp = vpa_)
</code></pre>

<p><img src="/public/img/2012-08-22-ggplot-inset-map/unnamed-chunk-11-1.png" alt="plot of chunk unnamed-chunk-11" /></p>

<p>Written in <a href="http://daringfireball.net/projects/markdown/">Markdown</a>, with help from <a href="http://yihui.name/knitr/">knitr</a>, and nice knitr highlighting/etc. in in <a href="http://rstudio.org/">RStudio</a>.</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2012/07/global-names-resolver/">
        Hitting the Global Names Resolver API
      </a>
    </h1>

    <span class="post-date">20 Jul 2012</span>

    <h2 id="example-of-using-the-global-names-resolver-api-to-check-species-names">Example of using the Global Names Resolver API to check species names</h2>

<hr />

<p>There are a number of options for resolution of taxonomic names. The <a href="http://tnrs.iplantcollaborative.org/">Taxonomic Name Resolution Service (TNRS)</a> comes to mind. There is a new service for taxonomic name resoultion called the <a href="http://resolver.globalnames.org/">Global Names Resolver</a>. They describe the service thusly “<em>Resolve lists of scientific names against known sources. This service parses incoming names, executes exact or fuzzy matching as required, and displays a confidence score for each match along with its identifier.</em>”.</p>

<hr />

<h2 id="load-required-packages">Load required packages</h2>

<h3 id="just-uncomment-the-code-to-use">Just uncomment the code to use.</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> <span class="c1"># If you don&#39;t have them already</span>
<span class="lineno">2</span> <span class="c1"># install.packages(c(&#39;RJSONIO&#39;,&#39;plyr&#39;,&#39;devtools&#39;)) require(devtools)</span>
<span class="lineno">3</span> <span class="c1"># install_github(&#39;taxize_&#39;,&#39;ropensci&#39;)</span>
<span class="lineno">4</span> <span class="kn">library</span><span class="p">(</span>RJSONIO<span class="p">)</span>
<span class="lineno">5</span> <span class="kn">library</span><span class="p">(</span>plyr<span class="p">)</span>
<span class="lineno">6</span> <span class="kn">library</span><span class="p">(</span>taxize<span class="p">)</span></code></pre></figure>

<h2 id="get-the-data-sources-available">Get the data sources available</h2>

<h3 id="get-just-ids-and-names-of-sources-in-a-dataframe">Get just id’s and names of sources in a data.frame</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> <span class="kp">tail</span><span class="p">(</span>gnr_datasources<span class="p">(</span>todf <span class="o">=</span> <span class="bp">T</span><span class="p">))</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">##     id                                title
## 82 164                            BioLib.cz
## 83 165 Tropicos - Missouri Botanical Garden
## 84 166                                nlbif
## 85 167                                 IPNI
## 86 168              Index to Organism Names
## 87 169                        uBio NameBank</code></pre></figure>

<hr />

<h3 id="give-me-the-id-for-eol-encyclopedia-of-life">Give me the id for EOL (Encyclopedia of Life)</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> out <span class="o">&lt;-</span> gnr_datasources<span class="p">(</span>todf <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="lineno">2</span> out<span class="p">[</span>out<span class="o">$</span>title <span class="o">==</span> <span class="s">&quot;EOL&quot;</span><span class="p">,</span> <span class="s">&quot;id&quot;</span><span class="p">]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## [1] 12</code></pre></figure>

<hr />

<h3 id="fuzzy-search-for-sources-with-the-word-zoo">Fuzzy search for sources with the word “zoo”</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> out <span class="o">&lt;-</span> gnr_datasources<span class="p">(</span>todf <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="lineno">2</span> outdf <span class="o">&lt;-</span> out<span class="p">[</span><span class="kp">agrep</span><span class="p">(</span><span class="s">&quot;zoo&quot;</span><span class="p">,</span> out<span class="o">$</span>title<span class="p">,</span> ignore.case <span class="o">=</span> <span class="bp">T</span><span class="p">),</span> <span class="p">]</span>
<span class="lineno">3</span> outdf<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,</span> <span class="p">]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">##     id             title
## 20 100 Mushroom Observer
## 25 105           ZooKeys</code></pre></figure>

<h2 id="resolve-some-names">Resolve some names</h2>

<h3 id="search-for-helianthus-annuus-and-homo-sapiens-return-a-dataframe">Search for <em>Helianthus annuus</em> and <em>Homo sapiens</em>, return a data.frame</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> gnr<span class="p">(</span>names <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;Helianthus annuus&quot;</span><span class="p">,</span> <span class="s">&quot;Homo sapiens&quot;</span><span class="p">),</span> returndf <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,</span> <span class="p">]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">##   data_source_id    submitted_name       name_string score    title
## 1              4 Helianthus annuus Helianthus annuus 0.988     NCBI
## 3             10 Helianthus annuus Helianthus annuus 0.988 Freebase</code></pre></figure>

<hr />

<h3 id="search-for-the-same-species-with-only-using-data-source-12-ie-eol">Search for the same species, with only using data source 12 (i.e., EOL)</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> gnr<span class="p">(</span>names <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;Helianthus annuus&quot;</span><span class="p">,</span> <span class="s">&quot;Homo sapiens&quot;</span><span class="p">),</span> data_source_ids <span class="o">=</span> <span class="s">&quot;12&quot;</span><span class="p">,</span> 
<span class="lineno">2</span>     returndf <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">##   data_source_id    submitted_name       name_string score title
## 1             12 Helianthus annuus Helianthus annuus 0.988   EOL
## 2             12      Homo sapiens      Homo sapiens 0.988   EOL</code></pre></figure>

<h3 id="thats-it-have-fun-and-put-bugscommentsetc-herehttpsgithubcomropenscitaxizeissues">That’s it. Have fun! And put bugs/comments/etc. <a href="https://github.com/ropensci/taxize_/issues">here</a>.</h3>

<hr />

<h3 id="written-in-markdownhttpdaringfireballnetprojectsmarkdown-with-help-from-knitrhttpyihuinameknitr-and-nice-knitr-highlightingetc-in-in-rstudiohttprstudioorg">Written in <a href="http://daringfireball.net/projects/markdown/">Markdown</a>, with help from <a href="http://yihui.name/knitr/">knitr</a>, and nice knitr highlighting/etc. in in <a href="http://rstudio.org/">RStudio</a>.</h3>

<hr />

<h3 id="i-prepared-the-markdown-for-this-post-by">I prepared the markdown for this post by:</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno"> 1</span> KnitPost <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>input<span class="p">,</span> base.url <span class="o">=</span> <span class="s">&quot;/&quot;</span><span class="p">)</span> <span class="p">{</span>
<span class="lineno"> 2</span>     <span class="kn">require</span><span class="p">(</span>knitr<span class="p">)</span>
<span class="lineno"> 3</span>     opts_knit<span class="o">$</span>set<span class="p">(</span>base.url <span class="o">=</span> base.url<span class="p">)</span>
<span class="lineno"> 4</span>     fig.path <span class="o">&lt;-</span> <span class="kp">paste0</span><span class="p">(</span><span class="s">&quot;img/&quot;</span><span class="p">,</span> <span class="kp">sub</span><span class="p">(</span><span class="s">&quot;.Rmd$&quot;</span><span class="p">,</span> <span class="s">&quot;&quot;</span><span class="p">,</span> <span class="kp">basename</span><span class="p">(</span>input<span class="p">)),</span> <span class="s">&quot;/&quot;</span><span class="p">)</span>
<span class="lineno"> 5</span>     opts_chunk<span class="o">$</span>set<span class="p">(</span>fig.path <span class="o">=</span> fig.path<span class="p">)</span>
<span class="lineno"> 6</span>     opts_chunk<span class="o">$</span>set<span class="p">(</span>fig.cap <span class="o">=</span> <span class="s">&quot;center&quot;</span><span class="p">)</span>
<span class="lineno"> 7</span>     render_jekyll<span class="p">()</span>
<span class="lineno"> 8</span>     knit<span class="p">(</span>input<span class="p">,</span> envir <span class="o">=</span> <span class="kp">parent.frame</span><span class="p">())</span>
<span class="lineno"> 9</span> <span class="p">}</span>
<span class="lineno">10</span> <span class="kp">setwd</span><span class="p">(</span><span class="s">&quot;/path/to/_posts&quot;</span><span class="p">)</span>
<span class="lineno">11</span> KnitPost<span class="p">(</span><span class="s">&quot;/path/to/postfile.Rmd&quot;</span><span class="p">)</span></code></pre></figure>

<p>from <a href="http://jfisher-usgs.github.com/r/2012/07/03/knitr-jekyll/">jfisher</a>.</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2012/06/recent-r-eeb-packages/">
        Recent R packages for ecology and evolution
      </a>
    </h1>

    <span class="post-date">14 Jun 2012</span>

    <p>Many R packages/tools have come out recently for doing ecology and evolution. All of the below were described in Methods in Ecology and Evolution, except for spider, which came out in <a href="http://onlinelibrary.wiley.com/journal/10.1111/(ISSN)1755-0998">Molecular Ecology Resources</a>. Here are some highlights.</p>

<ul>
  <li>mvabund <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2012.00190.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/mvabund/index.html">get R pkg</a>
    <ul>
      <li>Model-based analysis of multivariate abundance data. Visualising data, fitting predictive models, checking assumptions, hypothesis testing.</li>
    </ul>
  </li>
  <li>popdemo <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2012.00222.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/popdemo/index.html">get R pkg</a>
    <ul>
      <li>Population demography using projection matrix analysis.</li>
    </ul>
  </li>
  <li>motmot <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00132.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/motmot/index.html">get R pkg</a>
    <ul>
      <li>Models of trait macroevolution on trees</li>
    </ul>
  </li>
  <li>spider <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2011.03108.x/abstract?deniedAccessCustomisedMessage=&amp;userIsAuthenticated=false">paper</a> - <a href="http://cran.r-project.org/web/packages/spider/index.html">get R pkg</a>
    <ul>
      <li>Analysis of species identity and evolution, with particular reference to DNA barcoding</li>
    </ul>
  </li>
  <li>BaSTA <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2012.00186.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/BaSTA/index.html">get R pkg</a>
    <ul>
      <li>Bayesian estimation of age-specific survival from incomplete mark–recapture/recovery data with covariates</li>
    </ul>
  </li>
  <li>abc <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00179.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/abc/index.html">get R pkg</a>
    <ul>
      <li>Approximate Bayesian Computation (ABC)</li>
    </ul>
  </li>
  <li>RNetLogo <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00180.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/RNetLogo/index.html">get R pkg</a>
    <ul>
      <li>Running and exploring individual-based models implemented in NetLogo</li>
    </ul>
  </li>
  <li>phytools <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00169.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/phytools/index.html">get R pkg</a>
    <ul>
      <li>Tools for phylogenetic comparative biology</li>
    </ul>
  </li>
  <li>smatr <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00153.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/smatr/index.html">get R pkg</a>
    <ul>
      <li>Estimation and inference about allometric lines</li>
    </ul>
  </li>
  <li>RBrownie <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00112.x/abstract">paper</a> - <a href="http://www.brianomeara.info/tutorials/brownie">get R pkg ?</a>
    <ul>
      <li>Testing hypotheses about rates of evolutionary change</li>
    </ul>
  </li>
  <li>polytomy resolver <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00103.x/abstract">paper</a> - <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00103.x/suppinfo">get R pkg</a>
    <ul>
      <li>Resolve polytomies on dated phylogenies with their R scripts [here][].</li>
    </ul>
  </li>
</ul>

<p>And a cool tool came out for the Python programming language.</p>

<ul>
  <li>NichePy <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00184.x/abstract">paper</a> - <a href="https://github.com/bastodian/NichePy">get python</a>
    <ul>
      <li>Modular tools for estimating the similarity of ecological niche and species distribution models</li>
    </ul>
  </li>
</ul>


  </div>
  
</div>

<!-- Pagination links -->
<div class="pagination">
  
    <a href="/page32" class="older">Older</a>
  
  
    
      <a href="/page30" class="newer">Newer</a>
    
  
</div>

    </div>

    <!-- for bootstrap tooltips -->
    <script type="text/javascript">
      $("[data-toggle=\"tooltip\"]").tooltip();
    </script>

  </body>

  <footer>
  <!-- Disqus code -->
  <script type="text/javascript">
      /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
      var disqus_shortname = 'recology'; // required: replace example with your forum shortname

      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function () {
          var s = document.createElement('script'); s.async = true;
          s.type = 'text/javascript';
          s.src = 'http://' + disqus_shortname + '.disqus.com/count.js';
          (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
      }());
  </script>

  <!-- google analytics -->
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-63197374-1', 'auto');
    ga('send', 'pageview');
  </script>
</footer>

</html>
