<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <title>
    
    Recology, R/etc.
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:300,400italic,400,600,700|Abril+Fatface">

  <link rel="stylesheet" href="/public/css/bootstrap/css/bootstrap.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/favicon.ico">
  <link rel="shortcut icon" href="/public/favicon.ico">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css" rel="stylesheet">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body class="theme-base-0f layout-reverse">

    <header class="masthead">
      <div class="masthead-inner">
        <h1>Recology</h1>
        <!-- <h1> <a href="http://recology.info/">Recology</a></h1> -->
        <p class="lead">R/etc.</p>

        <div class="colophon">
          <ul class="colophon-links">
            <li>
              <a href="/"><i class="fa fa-home fa-lg"></i></a>&nbsp;
              <a href="/about"><i class="fa fa-info-circle fa-lg"></i></a>&nbsp;
              <a href="/archives"><i class="fa fa-archive fa-lg"></i></a>&nbsp;
              <a href="/rresources"><i class="fa fa-book fa-lg"></i></a>&nbsp;
              <a href="http://rforcats.net/" rel><i class="fa fa-graduation-cap fa-lg"></i></a>&nbsp;
              <a href="/feed.xml"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
              <a href="https://twitter.com/sckottie"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
              <a href="/fork"><i class="fa fa-spinner fa-lg"></i></a>
            </li>
          </ul>
          <!-- <small><a href="https://github.com/mdo/hyde">Hyde</a> from <a href="https://twitter.com/mdo" target="_blank">@mdo</a>.</small> -->
        </div>
      </div>
    </header>

    <div class="content container">
      <div class="posts">
  <a style="float:right;" href="/archives" data-toggle="tooltip" data-placement="bottom" title="Archives"><i class="fa fa-archive fa-lg"></i></a>
  <a style="float:right;" href="/tags"><i class="fa fa-tags fa-lg"></i></a>&nbsp;
  
  <div class="post">
    <h1>
      <a href="/2011/08/rnpn-r-interface-for-national-phenology/">
        rnpn: An R interface for the National Phenology Network
      </a>
    </h1>

    <span class="post-date">31 Aug 2011</span>

    <p>The team at <a href="http://ropensci.org/">rOpenSci</a> and I have been working on a wrapper for the <a href="http://www.usanpn.org/">USA National Phenology Network</a> API. The following is a demo of some of the current possibilities. We will have more functions down the road. Get the publicly available code, and contribute, at Github <a href="https://github.com/ropensci/rnpn">here</a>. If you try this out look at the <a href="https://github.com/ropensci/rnpn/blob/master/DESCRIPTION">Description file </a>for the required R packages to run rnpn. Let us know at Github (<a href="https://github.com/ropensci">here</a>) or at our website  <a href="http://ropensci.org/">http://ropensci.org/</a>, or in the comments below, or on twitter (@rOpenSci), what use cases you would like to see with the rnpn package.<br /><br /><b><u>Method and demo of each</u></b>:<br /><i><b>Get observations for species by day</b></i><br /><i><span class="Apple-style-span" style="font-style: normal;">From the documentation: “</span></i><span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function will return a list of species, containing all the dates which observations were made about the species, and a count of the number of such observations made on that date.</span><i><span class="Apple-style-span" style="font-style: normal;">”</span></i><br /><br /><span class="Apple-style-span" style="font-size: x-small;">#### Note, the data below is truncated for blogging brevity…</span><br />&lt;div style="overflow: auto;"&gt;&lt;div class="geshifilter"&gt;&lt;pre class="r geshifilter-R" style="font-family: monospace;"&gt;&gt; getobsspbyday<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">2</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: blue;">‘2008-01-01’</span><span style="color: #339933;">,</span> <span style="color: blue;">‘2011-12-31’</span><span style="color: #666666;">) # Searched for species 1 and 2 from Jan 1, 2008 to Dec 31, 2011</span><br />          <a href="http://inside-r.org/r-doc/base/date"><span style="color: #003399; font-weight: bold;">date</span></a> count   species<br /><span style="color: #cc66cc;">1</span>   <span style="color: #cc66cc;">2009</span>-03-08     <span style="color: #cc66cc;">2</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">2</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">15</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">3</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">22</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">4</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">24</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">5</span>   <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">26</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">6</span>   <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">17</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">7</span>   <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">24</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">8</span>   <span style="color: #cc66cc;">2009</span>-05-<span style="color: #cc66cc;">12</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">9</span>   <span style="color: #cc66cc;">2009</span>-05-<span style="color: #cc66cc;">20</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">10</span>  <span style="color: #cc66cc;">2009</span>-<span style="color: #cc66cc;">11</span>-<span style="color: #cc66cc;">24</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">11</span>  <span style="color: #cc66cc;">2009</span>-<span style="color: #cc66cc;">12</span>-07     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">12</span>  <span style="color: #cc66cc;">2010</span>-01-<span style="color: #cc66cc;">18</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">13</span>  <span style="color: #cc66cc;">2010</span>-01-<span style="color: #cc66cc;">23</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">62</span>  <span style="color: #cc66cc;">2011</span>-05-<span style="color: #cc66cc;">29</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">63</span>  <span style="color: #cc66cc;">2011</span>-06-<span style="color: #cc66cc;">27</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">64</span>  <span style="color: #cc66cc;">2011</span>-06-<span style="color: #cc66cc;">30</span>     <span style="color: #cc66cc;">2</span> species <span style="color: #cc66cc;">1</span><br /><span style="color: #cc66cc;">65</span>  <span style="color: #cc66cc;">2009</span>-03-<span style="color: #cc66cc;">17</span>     <span style="color: #cc66cc;">1</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">66</span>  <span style="color: #cc66cc;">2009</span>-04-03     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">67</span>  <span style="color: #cc66cc;">2009</span>-04-05     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">68</span>  <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">10</span>     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span><br /><span style="color: #cc66cc;">69</span>  <span style="color: #cc66cc;">2009</span>-04-<span style="color: #cc66cc;">17</span>     <span style="color: #cc66cc;">3</span> species <span style="color: #cc66cc;">2</span>&lt;/pre&gt;&lt;/div&gt;&lt;/div&gt;<br /><br /><br /><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><b>Get individuals at specific stations</b></i><br />From the documentation: “<span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function returns all of the individuals at a series of stations.</span>“<br />&lt;div&gt;<br />&lt;div style="overflow-x: auto; overflow-y: auto;"&gt;&lt;div class="geshifilter"&gt;&lt;pre class="r geshifilter-R" style="font-family: monospace;"&gt;&gt; getindsatstations<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">507</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">523</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span> <span style="color: #666666;"># Searched for any individuals at stations 507 and 523</span><br />   individual_id individual_name species_id kingdom<br /><span style="color: #cc66cc;">1</span>           <span style="color: #cc66cc;">1200</span>         dogwood         <span style="color: #cc66cc;">12</span> Plantae<br /><span style="color: #cc66cc;">2</span>           <span style="color: #cc66cc;">1197</span>    purple lilac         <span style="color: #cc66cc;">36</span> Plantae<br /><span style="color: #cc66cc;">3</span>           <span style="color: #cc66cc;">1193</span>         white <a href="http://inside-r.org/r-doc/base/t"><span style="color: #003399; font-weight: bold;">t</span></a>         <span style="color: #cc66cc;">38</span> Plantae<br /><span style="color: #cc66cc;">4</span>           <span style="color: #cc66cc;">3569</span>     forsythia-<span style="color: #cc66cc;">1</span>         <span style="color: #cc66cc;">73</span> Plantae<br /><span style="color: #cc66cc;">5</span>           <span style="color: #cc66cc;">1206</span>            jack        <span style="color: #cc66cc;">150</span> Plantae<br /><span style="color: #cc66cc;">6</span>           <span style="color: #cc66cc;">1199</span>      trout lily        <span style="color: #cc66cc;">161</span> Plantae<br /><span style="color: #cc66cc;">7</span>           <span style="color: #cc66cc;">1198</span>           dandy        <span style="color: #cc66cc;">189</span> Plantae<br /><span style="color: #cc66cc;">8</span>           <span style="color: #cc66cc;">1192</span>           red <a href="http://inside-r.org/r-doc/base/t"><span style="color: #003399; font-weight: bold;">t</span></a>        <span style="color: #cc66cc;">192</span> Plantae<br /><span style="color: #cc66cc;">9</span>           <span style="color: #cc66cc;">1710</span>    common lilac         <span style="color: #cc66cc;">36</span> Plantae<br /><span style="color: #cc66cc;">10</span>          <span style="color: #cc66cc;">1711</span>  common lilac <span style="color: #cc66cc;">2</span>         <span style="color: #cc66cc;">36</span> Plantae<br /><span style="color: #cc66cc;">11</span>          <span style="color: #cc66cc;">1712</span>       dandelion        <span style="color: #cc66cc;">189</span> Plantae&lt;/pre&gt;&lt;/div&gt;&lt;/div&gt;<br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><b>Get individuals of species at stations</b></i><br />From the documentation: “<span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function will return a list of all the individuals, which are members of a species, among  any number of stations.</span>”&lt;/div&gt;&lt;div&gt;<br />&lt;div style="overflow-x: auto; overflow-y: auto;"&gt;&lt;div class="geshifilter"&gt;&lt;pre class="r geshifilter-R" style="font-family: monospace;"&gt;&gt; getindspatstations<span style="color: #009900;">(</span><span style="color: #cc66cc;">35</span><span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">60</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">259</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">2009</span><span style="color: #009900;">)</span>  <span style="color: #666666;"># Search for individuals of species 35 at stations 60 and 259 in year 2009</span><br />  individual_id individual_name number_observations<br /><span style="color: #cc66cc;">1</span>          <span style="color: #cc66cc;">1715</span>            west                   <span style="color: #cc66cc;">5</span><br /><span style="color: #cc66cc;">2</span>          <span style="color: #cc66cc;">1716</span>            east                   <span style="color: #cc66cc;">5</span>&lt;/pre&gt;&lt;/div&gt;&lt;/div&gt;<br /><br /><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><br /></i><br /><i style="font-style: italic;"><b>Get observation associated with particular observation</b></i><br /><span class="Apple-style-span"><span class="Apple-style-span" style="font-style: normal;">From the documentation:</span><span class="Apple-style-span" style="font-family: Arial; font-size: x-small;"><span class="Apple-style-span" style="white-space: pre-wrap;"><span class="Apple-style-span" style="font-style: normal;"> </span>”</span></span></span><span class="Apple-style-span" style="font-family: Arial; font-size: 13px; white-space: pre-wrap;">This function will return the comment associated with a particular observation.</span><span class="Apple-style-span"><span class="Apple-style-span" style="font-family: Arial; font-size: x-small;"><span class="Apple-style-span" style="white-space: pre-wrap;">”</span></span></span><br /><span class="Apple-style-span" style="font-family: Arial; font-size: x-small;"><span class="Apple-style-span" style="white-space: pre-wrap;"></span></span><br /><span class="Apple-style-span" style="font-family: monospace; white-space: pre;">&gt; getobscomm<span style="color: #009900;">(</span><span style="color: #cc66cc;">1938</span><span style="color: #009900;">)</span> <span style="color: #666666;"># The observation for observation number 1938</span></span><br />&lt;div style="overflow-x: auto; overflow-y: auto;"&gt;&lt;div class="geshifilter"&gt;&lt;pre class="r geshifilter-R" style="font-family: monospace;"&gt;$observation_comment<br /><span style="color: #009900;">[</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span> <span style="color: blue;">“some lower branches are bare”</span>&lt;/pre&gt;&lt;/div&gt;&lt;/div&gt;&lt;/div&gt;</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2011/08/tenure-track-position-in-systematics-at/">
        Tenure track position in systematics at the University of Vermont
      </a>
    </h1>

    <span class="post-date">22 Aug 2011</span>

    <p><span class="Apple-style-span" style="background-color: white; color: #4f4f4f; font-family: Arial, Tahoma, Helvetica, FreeSans, sans-serif; font-size: 15px; line-height: 20px;"></span><br /><center style="text-align: left;">There is an awesome position opening up for an assistant professor in systematics at the University of Vermont. Below is the announcement, and see the <a href="http://currentecology.blogspot.com/2011/08/tenure-track-position-in-systematics-at.html">original post</a> at the <a href="http://currentecology.blogspot.com/">Distributed Ecology blog</a>. Why is this related to R? One can do a lot of systematics work in R, including retrieving scientific collections data through an upcoming package handshaking with <a href="http://vertnet.org/index.php">VertNet</a> (part of the <a href="http://ropensci.org/">rOpenSci</a> project), managing large data sets, retrieval of GenBank data through the ape package (see fxn read.genbank), phylogenetic reconstruction and analysis, and more. So I am sure a systematist with R ninja skills will surely have a head up on the rest of the field.&nbsp;</center><center style="text-align: left;"><br /></center><center style="text-align: left;"><br /></center><center><b><br /></b></center><center><b>Assistant Professor in Systematics</b></center><br /><br />Department of Biology<br />University of Vermont<br />Burlington, Vermont<br /><br />The Department of Biology of the University of Vermont seeks applications for a tenure- track Assistant Professor position in Systematics and Evolutionary Biology of arthropods, especially insects. The position will be open in the fall of 2012. The successful candidate will have expertise in classical and molecular systematics, including analysis of complex data sets. Candidates pursuing phylogenomics and innovative methods in bioinformatics in combination with taxonomy are especially encouraged to apply. Department information at:<a href="http://www.uvm.edu/~biology/" style="color: #0499fa; text-decoration: none;"> http://www.uvm.edu/~biology/</a>.<br /><br /><br />All applicants are expected to: 1) hold a Ph.D. degree in relevant disciplines and have two or more years of postdoctoral experience; 2) develop a competitively funded research program; 3) teach undergraduate courses (chosen from among general biology, evolution, systematic entomology, and others in the candidate’s expertise); 4) teach, mentor and advise undergraduate and graduate students; and 5) oversee a natural history collection of historic significance.<br /><br />Candidates must apply online:<a href="http://www.uvmjobs.com/" style="color: #0499fa; text-decoration: none;"> http://www.uvmjobs.com/</a>. On left see “Search Postings” then find “Biology” under “HCM Department” then posting 0040090 (first posting). Sorry, but we cannot supply the direct link because it will time out.<br /><br />Attach a cover letter with a statement of research focus and teaching interests (one document), a curriculum vitae, representative publications, and the contact information of three references.<br /><br />Review of applications will begin on September 15, 2011, and will continue until the position is filled. Questions and up to three additional publications may be directed to Dr. Jos. J. Schall: jschall@uvm.edu.<br /><br />The University of Vermont recently identified several “Spires of Excellence” in which it will strategically focus institutional investments and growth over the next several years. One spire associated with the position is Complex Systems. Candidates whose research interests align with this spire are especially encouraged to apply<a href="http://www.uvm.edu/~tri/" style="color: #0499fa; text-decoration: none;">http://www.uvm.edu/~tri/</a>.<br />The University seeks faculty who can contribute to the diversity and excellence of the academic community through their research, teaching, and/or service. Applicants are requested to include in their cover letter information about how they will further this goal. The University of Vermont is an Affirmative Action/Equal Opportunity employer. The Department is committed to increasing faculty diversity and welcomes applications from women, underrepresented ethnic, racial and cultural groups, and from people with disabilities.</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2011/08/thursday-at-esa11/">
        Thursday at #ESA11
      </a>
    </h1>

    <span class="post-date">12 Aug 2011</span>

    <p>Interesting talks/posters:&lt;div&gt;<br />Richard Lankau presented research on trade-offs and competitive ability. He suggests that during range expansion selection for increased intraspecific competitive ability in older populations leads to loss of traits for interspecific competitive traits due to trade-offs between these traits. <br /><br /><br />Ellner emphatically states that rapid evolution DOES matter for ecological responses, and longer-term evolutionary patterns as well. [His paper on the talk he was giving came out prior to his talk, which he pointed out, good form sir]<br /><br /><br />Lauren Sullivan gave an interesting talk on bottom up and top down effects on plant reproduction in one site of a huge network of sites doing similar nutrient and herbivory manipulations around the globe - NutNet (go here: <a href="http://nutnet.science.oregonstate.edu/">http://nutnet.science.oregonstate.edu/</a>). <br /><br /><br />Laura Prugh shows in California that the engineering effects (i.e., the mounds that they make) of giant kangaroo rats are more important for the associated food web than the species interaction effects (the proxy used was just density of rats). <br /><br /><br />Kristy Deiner suggests that chironomids are more phylogenetic similar in lakes with stocked fish relative to fishless lakes, in high elevation lakes in the Sierra Nevada. She used barcode data to generate her phylogeny of chironomids. If you have barcode data and want to search BOLD Systems site, one option is doing it from R using rbold, a package under development at <a href="http://ropensci.org/">rOpenSci</a> (code at <a href="https://github.com/sckott/rbold">Github</a>). <br /><br /><br />Jessica Gurevitch presented a large working group’s methods/approach to a set of reviews on invasion biology. We didn’t get to see a lot of results from this work, but I personally was glad to see her explaining to a packed room the utility of meta-analysis, and comparing to the medical field in which meta-analysis is sort of the gold standard by which to draw conclusions. <br /><br /><br />Following Jessica, Jason Fridley told us about the Evolutionary Imbalance Hypothesis (EIH) (see my notes <a href="http://schamber.wordpress.com/lab-notebook-2/">here</a>). He posed the problem of, when two biotas come together, what determines which species are retained in this new community and which species are left out. He listed a litany of traits/responses to measure to get at this problem, but suggested that with a little bit of “desktop ecology”, we could simply ask: Is the invasability of X region related to the phylogenetic diversity of that region? In three destination regions (Eastern Deciduous Forests, Mediterranean California, and the Czech Republic) out of four there was a positive relationship between proportion of invasive plant species in a source region and the phylogenetic diversity of the source regions. &lt;ul&gt;&lt;/ul&gt;&lt;/div&gt;</p>

  </div>
  
</div>

<!-- Pagination links -->
<div class="pagination">
  
    <a href="/page47" class="older">Older</a>
  
  
    
      <a href="/page45" class="newer">Newer</a>
    
  
</div>

    </div>

    <!-- for bootstrap tooltips -->
    <script type="text/javascript">
      $("[data-toggle=\"tooltip\"]").tooltip();
    </script>

  </body>

  <footer>
  <!-- Disqus code -->
  <script type="text/javascript">
      /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
      var disqus_shortname = 'recology'; // required: replace example with your forum shortname

      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function () {
          var s = document.createElement('script'); s.async = true;
          s.type = 'text/javascript';
          s.src = 'http://' + disqus_shortname + '.disqus.com/count.js';
          (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
      }());
  </script>

  <!-- google analytics -->
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-63197374-1', 'auto');
    ga('send', 'pageview');
  </script>
</footer>

</html>
