<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <title>
    
    Recology, R/etc.
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:300,400italic,400,600,700|Abril+Fatface">

  <link rel="stylesheet" href="/public/css/bootstrap/css/bootstrap.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/favicon.ico">
  <link rel="shortcut icon" href="/public/favicon.ico">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css" rel="stylesheet">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body class="theme-base-0f layout-reverse">

    <header class="masthead">
      <div class="masthead-inner">
        <h1>Recology</h1>
        <!-- <h1> <a href="http://recology.info/">Recology</a></h1> -->
        <p class="lead">R/etc.</p>

        <div class="colophon">
          <ul class="colophon-links">
            <li>
              <a href="/"><i class="fa fa-home fa-lg"></i></a>&nbsp;
              <a href="/about"><i class="fa fa-info-circle fa-lg"></i></a>&nbsp;
              <a href="/archives"><i class="fa fa-archive fa-lg"></i></a>&nbsp;
              <a href="/rresources"><i class="fa fa-book fa-lg"></i></a>&nbsp;
              <a href="http://rforcats.net/" rel><i class="fa fa-graduation-cap fa-lg"></i></a>&nbsp;
              <a href="/feed.xml"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
              <a href="https://twitter.com/sckottie"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
              <a href="/fork"><i class="fa fa-spinner fa-lg"></i></a>
            </li>
          </ul>
          <!-- <small><a href="https://github.com/mdo/hyde">Hyde</a> from <a href="https://twitter.com/mdo" target="_blank">@mdo</a>.</small> -->
        </div>
      </div>
    </header>

    <div class="content container">
      <div class="posts">
  <a style="float:right;" href="/archives" data-toggle="tooltip" data-placement="bottom" title="Archives"><i class="fa fa-archive fa-lg"></i></a>
  <a style="float:right;" href="/tags"><i class="fa fa-tags fa-lg"></i></a>&nbsp;
  
  <div class="post">
    <h1>
      <a href="/2011/07/crdata-vs-cloudnumbers/">
        CRdata vs. Cloudnumbers
      </a>
    </h1>

    <span class="post-date">14 Jul 2011</span>

    <p><a href="http://www.cloudnumbers.com/">Cloudnumbers</a> and <a href="http://crdata.org/">CRdata</a> are two new cloud computing services.<br /><br /><br />I tested the two services with a very simple script. The script simply creates a dataframe of 10000 numbers via rnorm, and assigns them to a factor of one of two levels (a or b). I then take the mean of the two factor levels with the aggregate function.<br /><br /><br />In CRdata you need to put in some extra code to format the output in a browser window. For example, the last line below needs to have ‘&lt;crdata_object&gt;’ on both sides of the output object so it can be rendered in a browser. And etc. for other things that one would print to a console. Whereas you don’t need this extra code for using Cloudnumbers.<br /><br />&lt;div style="overflow: auto;"&gt;&lt;div class="geshifilter"&gt;&lt;pre class="r geshifilter-R" style="font-family: monospace;"&gt; <br />dat &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>n = <a href="http://inside-r.org/r-doc/stats/rnorm"><span style="color: #003399; font-weight: bold;">rnorm</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">10000</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> p = <a href="http://inside-r.org/r-doc/base/rep"><span style="color: #003399; font-weight: bold;">rep</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: blue;">‘a’</span><span style="color: #339933;">,</span><span style="color: blue;">‘b’</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> each=<span style="color: #cc66cc;">5000</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /> <br />out &lt;- <a href="http://inside-r.org/r-doc/stats/aggregate"><span style="color: #003399; font-weight: bold;">aggregate</span></a><span style="color: #009900;">(</span>n ~ p<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/utils/data"><span style="color: #003399; font-weight: bold;">data</span></a> = dat<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/mean"><span style="color: #003399; font-weight: bold;">mean</span></a><span style="color: #009900;">)</span><br /> <br /><span style="color: #666666; font-style: italic;">#&lt;crdata_object&gt;out&lt;/crdata_object&gt;</span>&lt;/pre&gt;&lt;/div&gt;&lt;/div&gt;<br /><br />Here is a screenshot of the output from CRdata with the simple script above.<br /><img height="208" src="http://f.cl.ly/items/1D090q2N0Y3a410W262V/Screen%20shot%202011-07-14%20at%208.04.33%20AM.png" width="400" /><br />This simple script ran in about 20 seconds or so from starting the job to finishing. However, it seems like the only output option is html. Can this be right? This seems like a terrible only option.<br /><br /><br />In Cloudnumbers you have to start a workspace, upload your R code file.<br />Then, start a session…<br />choose your software platform…<br />choose packages (one at a time, very slow)…<br />then choose number of clusters, etc.<br />Then finally star the job.<br />Then it initializes, then finally you can open the console, and<br />Then from here it is like running R as you normally would, except on the web.<br /><br /><br /><u>Who wins (at least for our very minimal example above)</u><br /><br />&lt;ol&gt;&lt;li&gt;Speed of entire process (not just running code): CRdata&lt;/li&gt;&lt;li&gt;Ease of use: CRdata&lt;/li&gt;&lt;li&gt;Cost: CRdata (free only)&lt;/li&gt;&lt;li&gt;Least annoying: Cloudnumbers (you don’t have to add in extra code to run your own code)&lt;/li&gt;&lt;li&gt;Opensource: CRdata (you can use publicly available code on the site)&lt;/li&gt;&lt;li&gt;Long-term use: Cloudnumbers (more powerful, flexible, etc.)&lt;/li&gt;&lt;/ol&gt;&lt;div&gt;<br />&lt;/div&gt;&lt;div&gt;I imagine Cloudnumbers could be faster for larger jobs, but you would have to pay for the speed of course. &lt;/div&gt;&lt;div&gt;<br />&lt;/div&gt;&lt;div&gt;What I really want to see is a cloud computing service that accepts code directly run from R or RStudio. Hmmm…that would be so tasty indeed. I think<a href="http://cloudnumbers.zendesk.com/entries/20199198-using-an-external-terminal-ssh-console"> Cloudnumbers may be able to do this</a>, but haven’t tested it yet.  &lt;/div&gt;&lt;div&gt;<br />&lt;/div&gt;&lt;div&gt;Perhaps using the server version of RStudio along with Amazon’s EC2 is a better option than both of these. See Karthik Ram’s <a href="http://inundata.org/2011/03/30/r-ec2-rstudio-server/">post</a> about using RStudio server along with Amazon’s EC2. Even just running RStudio server on your Unbuntu machine or virtual machine is a pretty cool option, even without EC2 (works like a charm on my Parallels Ubuntu vm on my Mac). &lt;/div&gt;</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2011/06/rbold-an-r-interface-for-bold-systems-barcode-repository/">
        rbold: An R Interface for Bold Systems barcode repository
      </a>
    </h1>

    <span class="post-date">28 Jun 2011</span>

    <p>Have you ever wanted to <a href="http://services.boldsystems.org/index.php?page=1_esearch&amp;status=">search</a> and <a href="http://services.boldsystems.org/index.php?page=2_efetch&amp;status=">fetch</a> barcode data from <a href="http://www.boldsystems.org/views/login.php">Bold Systems</a>?<br /><br />I am developing functions to interface with Bold from R. I just started, but hopefully folks will find it useful.<br /><br />The code is at Github <a href="https://github.com/ropensci/rbold">here</a>. The two functions are still very buggy, so please bring up issues below, or in the Issues area on Github. For example, some searches work and other similar searches don’t. Apologies in advance for the bugs.<br /><br />Below is a screenshot of an example query using function getsampleids to get barcode identifiers for specimens. You can then use getseqs function to grab barcode data for specific specimens or many specimens.<br /><img alt="Screen shot 2011-06-28 at 9.24.00 AM.png" height="267" src="http://cl.ly/1V1y1Q1A0t062F2y2308/Screen_shot_2011-06-28_at_9.24.00_AM.png" width="400" /></p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2011/06/ievobio-2011-synopsis/">
        iEvoBio 2011 Synopsis
      </a>
    </h1>

    <span class="post-date">22 Jun 2011</span>

    <p>We just wrapped up the <a href="http://ievobio.org/">2011 iEvoBio meeting</a>. It was awesome! If you didn’t go this year or last year, definitely think about going next year.<br />&lt;div&gt;<br />&lt;/div&gt;&lt;div&gt;<br />&lt;/div&gt;&lt;div&gt;Here is a list of the cool projects that were discussed at the meeting (apologies if I left some out):&lt;/div&gt;&lt;div&gt;&lt;ol&gt;&lt;li&gt;<a href="http://vistrails.org/index.php/Main_Page">Vistrails</a>: workflow tool, awesome project by Claudio Silva&lt;/li&gt;&lt;li&gt;<a href="http://www.commplish.com/">Commplish</a>: purpose is to use via API’s, not with the web UI&lt;/li&gt;&lt;li&gt;<a href="http://phylopic.org/">Phylopic</a>: a database of life-form silouhettes, including an API for remote access, sweet!&lt;/li&gt;&lt;li&gt;<a href="http://gloome.tau.ac.il/">Gloome</a>&lt;/li&gt;&lt;li&gt;<a href="http://www.mappinglife.org/">MappingLife</a>: awesome geographic/etc data visualization interace on the web&lt;/li&gt;&lt;li&gt;<a href="http://bioinfolab.unl.edu/~canderson/SuiteMSA/">SuiteSMA</a>: visualizating multiple alignments&lt;/li&gt;&lt;li&gt;<a href="https://github.com/cboettig/treeBASE">treeBASE</a>: R interface to treebase, by <a href="http://www.carlboettiger.info/">Carl Boettiger</a>&lt;/li&gt;&lt;li&gt;<a href="http://vertnet.org/index.php">VertNet</a>: database for vertebrate natural history collections&lt;/li&gt;&lt;li&gt;<a href="http://sourceforge.net/projects/revbayes/">RevBayes</a>: revamp of MrBayes, with GUI, etc. &lt;/li&gt;&lt;li&gt;<a href="https://www.phenoscape.org/wiki/Main_Page">Phenoscape Knowledge Base</a>&lt;/li&gt;&lt;ul&gt;&lt;li&gt;Peter Midford lightning talk: talked about matching taxonomic and genetic data&lt;/li&gt;&lt;/ul&gt;&lt;li&gt;<a href="http://biscicol.blogspot.com/p/biscicol-prototype.html">BiSciCol</a>: biological science collections tracker&lt;/li&gt;&lt;li&gt;<a href="http://www.ontogrator.org/">Ontogrator</a> &lt;/li&gt;&lt;li&gt;<a href="http://tnrs.iplantcollaborative.org/">TNRS</a>: taxonomic name resolution service&lt;/li&gt;&lt;li&gt;<a href="http://www.boldsystems.org/views/login.php">Barcode of Life data systems</a>, and <a href="http://services.boldsystems.org/">remote access</a>&lt;/li&gt;&lt;li&gt;<a href="http://www.mooreabiocode.org/about">Moorea Biocode Project</a>&lt;/li&gt;&lt;li&gt;<a href="http://amarallab.mbl.edu/mirada/mirada.html">Microbial LTER’s data</a>&lt;/li&gt;&lt;li&gt;<a href="http://birdvis.org/">BirdVis</a>: interactive bird data visualization (Claudio Silva in collaboration with Cornell Lab of Ornithology)&lt;/li&gt;&lt;li&gt;<a href="http://www.crowdlabs.org/">Crowdlabs</a>: I think the site is down right now, another project by Claudio Silva&lt;/li&gt;&lt;li&gt;<a href="http://hydrodictyon.eeb.uconn.edu/projects/phycas/index.php/Phycas_Home">Phycas</a>: Bayesian phylogenetics, can you just call this from R?&lt;/li&gt;&lt;li&gt;RIP MrBayes!!!! replaced by RevBayes (see 9 above)&lt;/li&gt;&lt;li&gt;Slides of presentations will be at <a href="http://www.slideshare.net/event/ievobio11">Slideshare</a> (not all presentations up yet)          &lt;/li&gt;&lt;li&gt;A birds of a feather group I was involved in proposed an idea (TOL-o-matic) like Phylomatic, but of broader scope, for easy access and submission of trees, and perhaps even social (think just pushing a ‘SHARE’ button within PAUP, RevBayes, or other phylogenetics software)! &lt;/li&gt;&lt;li&gt;Synopses of Birds of a Feather discussion groups: <a href="http://piratepad.net/iEvoBio11-BoF-reportouts">http://piratepad.net/iEvoBio11-BoF-reportouts</a>&lt;/li&gt;&lt;/ol&gt;&lt;/div&gt;</p>

  </div>
  
</div>

<!-- Pagination links -->
<div class="pagination">
  
    <a href="/page49" class="older">Older</a>
  
  
    
      <a href="/page47" class="newer">Newer</a>
    
  
</div>

    </div>

    <!-- for bootstrap tooltips -->
    <script type="text/javascript">
      $("[data-toggle=\"tooltip\"]").tooltip();
    </script>

  </body>

  <footer>
  <!-- Disqus code -->
  <script type="text/javascript">
      /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
      var disqus_shortname = 'recology'; // required: replace example with your forum shortname

      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function () {
          var s = document.createElement('script'); s.async = true;
          s.type = 'text/javascript';
          s.src = 'http://' + disqus_shortname + '.disqus.com/count.js';
          (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
      }());
  </script>

  <!-- google analytics -->
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-63197374-1', 'auto');
    ga('send', 'pageview');
  </script>
</footer>

</html>
